--- a/src/main/java/org/bukkit/event/player/AsyncPlayerPreLoginEvent.java
+++ b/src/main/java/org/bukkit/event/player/AsyncPlayerPreLoginEvent.java
@@ -31,6 +_,7 @@
     private final InetAddress rawAddress;
     private final String hostname;
     private final boolean transferred;
+    private boolean skipLog; // Cheetah - Allow skipping log messages for canceled login events
     private Result result;
     private Component message;
     private PlayerProfile profile;
@@ -324,6 +_,16 @@
     public static HandlerList getHandlerList() {
         return HANDLER_LIST;
     }
+
+    // Cheetah start - Allow skipping log messages for canceled login events
+    public boolean isSkipLog() {
+        return this.skipLog;
+    }
+
+    public void setSkipLog(boolean skipLog) {
+        this.skipLog = skipLog;
+    }
+    // Cheetah end - Allow skipping log messages for canceled login events
 
     /**
      * Basic kick reasons for communicating to plugins
