--- a/src/main/java/org/bukkit/event/player/PlayerPreLoginEvent.java
+++ b/src/main/java/org/bukkit/event/player/PlayerPreLoginEvent.java
@@ -30,6 +_,7 @@
     private final String name;
     private final InetAddress ipAddress;
     private final UUID uniqueId;
+    private boolean skipLog; // Cheetah - Allow skipping log messages for canceled login events
 
     @Deprecated(since = "1.7.5")
     public PlayerPreLoginEvent(@NotNull final String name, @NotNull final InetAddress ipAddress) {
@@ -175,6 +_,16 @@
     public UUID getUniqueId() {
         return uniqueId;
     }
+
+    // Cheetah start - Allow skipping log messages for canceled login events
+    public boolean isSkipLog() {
+        return this.skipLog;
+    }
+
+    public void setSkipLog(boolean skipLog) {
+        this.skipLog = skipLog;
+    }
+    // Cheetah end
 
     @NotNull
     public static HandlerList getHandlerList() {
