--- a/net/minecraft/network/Connection.java
+++ b/net/minecraft/network/Connection.java
@@ -188,7 +_,7 @@
 
                     if (player != null) player.quitReason = org.bukkit.event.player.PlayerQuitEvent.QuitReason.ERRONEOUS_STATE; // Paper - Add API for quit reason
                     if (flag) {
-                        LOGGER.debug("Failed to sent packet", exception);
+                        Connection.LOGGER.warn("Failed to send to " + player.getName() + " packet", exception); // Cheetah - change log level to warn
                         boolean doesDisconnectExist = this.packetListener.protocol() != ConnectionProtocol.STATUS && this.packetListener.protocol() != ConnectionProtocol.HANDSHAKING; // Paper
                         if (this.getSending() == PacketFlow.CLIENTBOUND && doesDisconnectExist) { // Paper
                             Packet<?> packet = (Packet<?>)(this.sendLoginDisconnect
