--- a/net/minecraft/world/level/block/SnifferEggBlock.java
+++ b/net/minecraft/world/level/block/SnifferEggBlock.java
@@ -108,7 +_,7 @@
         int i = flag ? level.paperConfig().entities.sniffer.boostedHatchTime.or(BOOSTED_HATCH_TIME_TICKS) : level.paperConfig().entities.sniffer.hatchTime.or(REGULAR_HATCH_TIME_TICKS); // Paper
         int i1 = i / 3;
         level.gameEvent(GameEvent.BLOCK_PLACE, pos, GameEvent.Context.of(state));
-        level.scheduleTick(pos, this, i1 + level.random.nextInt(300));
+        level.scheduleTick(pos, this, i1 + level.random.nextInt(RANDOM_HATCH_OFFSET_TICKS));    // Cheetah - Using RANDOM_HATCH_OFFSET_TICKS instead of a magic number
     }
 
     @Override
