--- a/net/minecraft/world/level/block/TurtleEggBlock.java
+++ b/net/minecraft/world/level/block/TurtleEggBlock.java
@@ -108,7 +_,7 @@
     protected void randomTick(final BlockState state, final ServerLevel level, final BlockPos pos, final RandomSource random) {
         if (this.shouldUpdateHatchLevel(level, pos) && onSand(level, pos)) {
             int hatch = state.getValue(HATCH);
-            if (hatch < 2) {
+            if (hatch < MAX_HATCH_LEVEL) { // Cheetah - Using MAX_HATCH_LEVEL instead of a magic number
                 // CraftBukkit start - Call BlockGrowEvent
                 if (!org.bukkit.craftbukkit.event.CraftEventFactory.handleBlockGrowEvent(level, pos, state.setValue(HATCH, hatch + 1), Block.UPDATE_CLIENTS)) {
                     return;
