--- a/net/minecraft/world/level/block/TurtleEggBlock.java
+++ b/net/minecraft/world/level/block/TurtleEggBlock.java
@@ -108,7 +_,7 @@
     protected void randomTick(BlockState state, ServerLevel level, BlockPos pos, RandomSource random) {
         if (this.shouldUpdateHatchLevel(level, pos) && onSand(level, pos)) {
             int hatchValue = state.getValue(HATCH);
-            if (hatchValue < 2) {
+            if (hatchValue < MAX_HATCH_LEVEL) { // Cheetah - Using MAX_HATCH_LEVEL instead of a magic number
                 // CraftBukkit start - Call BlockGrowEvent
                 if (!org.bukkit.craftbukkit.event.CraftEventFactory.handleBlockGrowEvent(level, pos, state.setValue(HATCH, hatchValue + 1), 2)) {
                     return;
