--- a/net/minecraft/world/item/component/BlocksAttacks.java
+++ b/net/minecraft/world/item/component/BlocksAttacks.java
@@ -87,6 +_,7 @@
 
     public void disable(final ServerLevel level, final LivingEntity user, final float baseSeconds, final ItemStack blockingWith, LivingEntity attacker) { // Paper
         int cooldownTicks = this.disableBlockingForTicks(baseSeconds);
+        boolean playShieldBreakSound = true; // Cheetah - Suppress shield break sound if it was cancelled in the PlayerShieldDisableEvent
         if (cooldownTicks > 0) {
             if (user instanceof Player player) {
                 // Paper start
@@ -94,9 +_,15 @@
                 if (!shieldDisableEvent.callEvent()) return;
                 player.getCooldowns().addCooldown(blockingWith, shieldDisableEvent.getCooldown());
                 // Paper end
+                playShieldBreakSound = !shieldDisableEvent.isSoundCancelled(); // Cheetah - Suppress shield break sound if it was cancelled in the PlayerShieldDisableEvent
             }
 
             user.stopUsingItem();
+            // Cheetah start - Suppress shield break sound if it was cancelled in the PlayerShieldDisableEvent
+            if (!playShieldBreakSound) {
+                return;
+            }
+            // Cheetah end - Suppress shield break sound if it was cancelled in the PlayerShieldDisableEvent
             this.disableSound
                 .ifPresent(
                     sound -> level.playSound(
