--- a/src/main/java/com/destroystokyo/paper/profile/PaperMinecraftSessionService.java
+++ b/src/main/java/com/destroystokyo/paper/profile/PaperMinecraftSessionService.java
@@ -19,9 +_,19 @@
         super(servicesKeySet, proxy, environment);
     }
 
-    public @Nullable ProfileResult fetchProfile(GameProfile profile, final boolean requireSecure) {
+    // Cheetah start - Call PreFillProfile and FillProfileEvent
+    @Override
+    @Deprecated
+    public @Nullable ProfileResult fetchProfile(final UUID profileId, final boolean requireSecure) {
+        GameProfile profile = new GameProfile(
+            profileId,
+            ""
+        );
         CraftPlayerProfile playerProfile = (CraftPlayerProfile) CraftPlayerProfile.asBukkitCopy(profile);
-        new PreFillProfileEvent(playerProfile).callEvent();
+        final PreFillProfileEvent preFillProfileEvent = new PreFillProfileEvent(playerProfile);
+        preFillProfileEvent.callEvent();
+        playerProfile = (CraftPlayerProfile) preFillProfileEvent.getPlayerProfile();
+
         profile = playerProfile.getGameProfile();
         if (profile.properties().containsKey("textures")) {
             return new ProfileResult(profile, Collections.emptySet());
@@ -34,9 +_,5 @@
         }
         return result;
     }
-
-    @Override @Deprecated
-    public @Nullable ProfileResult fetchProfile(final UUID profileId, final boolean requireSecure) {
-        return super.fetchProfile(profileId, requireSecure);
-    }
+    // Cheetah end - Call PreFillProfile and FillProfileEvent
 }
