--- a/net/minecraft/world/level/levelgen/structure/structures/EndCityPieces.java
+++ b/net/minecraft/world/level/levelgen/structure/structures/EndCityPieces.java
@@ -427,7 +_,7 @@
                         sentry.setPos(position.getX() + 0.5, position.getY(), position.getZ() + 0.5);
                         level.addFreshEntity(sentry);
                     }
-                } else if (markerId.startsWith("Elytra")) {
+                } else if (markerId.startsWith("Elytra") && io.papermc.paper.configuration.GlobalConfiguration.get().misc.enableElytraInEndCities) { // Cheetah - Add option to disable elytra in end cities
                     ItemFrame itemFrame = new ItemFrame(level.getLevel(), position, this.placeSettings.getRotation().rotate(Direction.SOUTH));
                     itemFrame.setItem(new ItemStack(Items.ELYTRA), false);
                     level.addFreshEntity(itemFrame);
