--- a/net/minecraft/advancements/AdvancementRewards.java
+++ b/net/minecraft/advancements/AdvancementRewards.java
@@ -37,7 +_,7 @@
     public static final AdvancementRewards EMPTY = new AdvancementRewards(0, List.of(), List.of(), Optional.empty());
 
     public void grant(final ServerPlayer player) {
-        player.giveExperiencePoints(this.experience);
+        // Cheetah start - Add PlayerAdvancementRewardsGrantEvent
         ServerLevel level = player.level();
         MinecraftServer server = level.getServer();
         LootParams params = new LootParams.Builder(level)
@@ -46,37 +_,65 @@
             .create(LootContextParamSets.ADVANCEMENT_REWARD);
         boolean changes = false;
 
+        List<ItemStack> rewards = new java.util.ArrayList<>();
+
         for (ResourceKey<LootTable> lootTable : this.loot) {
-            for (ItemStack itemStack : server.reloadableRegistries().getLootTable(lootTable).getRandomItems(params)) {
-                if (player.addItem(itemStack)) {
-                    level.playSound(
-                        null,
-                        player.getX(),
-                        player.getY(),
-                        player.getZ(),
-                        SoundEvents.ITEM_PICKUP,
-                        SoundSource.PLAYERS,
-                        0.2F,
-                        ((player.getRandom().nextFloat() - player.getRandom().nextFloat()) * 0.7F + 1.0F) * 2.0F
-                    );
-                    changes = true;
-                } else {
-                    ItemEntity drop = player.drop(itemStack, false);
-                    if (drop != null) {
-                        drop.setNoPickUpDelay();
-                        drop.setTarget(player.getUUID());
+            rewards.addAll(server.reloadableRegistries().getLootTable(lootTable).getRandomItems(params));
+        }
+
+        net.gommehd.cheetah.event.entity.PlayerAdvancementRewardsGrantEvent event = new net.gommehd.cheetah.event.entity.PlayerAdvancementRewardsGrantEvent(
+            player.getBukkitEntity(),
+            recipes.stream().map(recipe -> org.bukkit.craftbukkit.util.CraftNamespacedKey.fromMinecraft(recipe.identifier())).collect(java.util.stream.Collectors.toList()),
+            rewards.stream().map(org.bukkit.craftbukkit.inventory.CraftItemStack::asCraftMirror).collect(java.util.stream.Collectors.toList()),
+            this.experience
+        );
+
+        org.bukkit.Bukkit.getPluginManager().callEvent(event);
+
+        if (!event.isCancelled()) {
+            rewards = event.getItems().stream().map(org.bukkit.craftbukkit.inventory.CraftItemStack::asNMSCopy).collect(java.util.stream.Collectors.toList());
+
+            if (event.getExperience() != 0) {
+                player.giveExperiencePoints(0);
+            }
+
+            if (!rewards.isEmpty()) {
+                for (ItemStack itemStack : rewards) {
+                    if (player.addItem(itemStack)) {
+                        level.playSound(
+                            null,
+                            player.getX(),
+                            player.getY(),
+                            player.getZ(),
+                            SoundEvents.ITEM_PICKUP,
+                            SoundSource.PLAYERS,
+                            0.2F,
+                            ((player.getRandom().nextFloat() - player.getRandom().nextFloat()) * 0.7F + 1.0F) * 2.0F
+                        );
+                        changes = true;
+                    } else {
+                        ItemEntity itemEntity = player.drop(itemStack, false);
+                        if (itemEntity != null) {
+                            itemEntity.setNoPickUpDelay();
+                            itemEntity.setTarget(player.getUUID());
+                        }
                     }
                 }
             }
         }
 
+        if (!event.getRecipes().isEmpty()) {
+            player.awardRecipesByKey(
+                event.getRecipes().stream()
+                    .map(key -> ResourceKey.create(net.minecraft.core.registries.Registries.RECIPE, org.bukkit.craftbukkit.util.CraftNamespacedKey.toMinecraft(key)))
+                    .toList()
+            );
+        }
+
         if (changes) {
             player.containerMenu.broadcastChanges();
         }
-
-        if (!this.recipes.isEmpty()) {
-            player.awardRecipesByKey(this.recipes);
-        }
+        // Cheetah end - Add PlayerAdvancementRewardsGrantEvent
 
         this.function
             .flatMap(function -> function.get(server.getFunctions()))
