--- a/net/minecraft/world/level/block/entity/ContainerOpenersCounter.java
+++ b/net/minecraft/world/level/block/entity/ContainerOpenersCounter.java
@@ -37,6 +_,12 @@
 
     protected abstract boolean isOwnContainer(Player player);
 
+    // Cheetah start - [MC-232968] Fixed spectator counting towards container view
+    private boolean isValidContainerViewer(Player player) {
+        return !player.isSpectator() && isOwnContainer(player);
+    }
+    // Cheetah end
+
     public void incrementOpeners(Player player, Level level, BlockPos pos, BlockState state) {
         int oldPower = Math.max(0, Math.min(15, this.openCount)); // CraftBukkit - Get power before new viewer is added
         int i = this.openCount++;
@@ -88,7 +_,7 @@
     private List<Player> getPlayersWithContainerOpen(Level level, BlockPos pos) {
         double d = this.maxInteractionRange + 4.0;
         AABB aabb = new AABB(pos).inflate(d);
-        return level.getEntities(EntityTypeTest.forClass(Player.class), aabb, this::isOwnContainer);
+        return level.getEntities(EntityTypeTest.forClass(Player.class), aabb, this::isValidContainerViewer); // Cheetah - [MC-232968] Fixed spectator counting towards container view
     }
 
     public void recheckOpeners(Level level, BlockPos pos, BlockState state) {
