--- a/net/minecraft/server/level/ServerPlayer.java
+++ b/net/minecraft/server/level/ServerPlayer.java
@@ -13,6 +_,7 @@
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import java.net.InetSocketAddress;
 import java.util.Collection;
+import java.util.Collections;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Optional;
@@ -831,8 +_,23 @@
         Entity camera = this.getCamera();
         if (camera != this) {
             if (camera.isAlive()) {
-                this.absSnapTo(camera.getX(), camera.getY(), camera.getZ(), camera.getYRot(), camera.getXRot());
-                this.level().getChunkSource().move(this);
+                // Cheetah start - Fix GLOBAL-1080
+                if (camera.level() != this.level()) {
+                    this.getBukkitEntity().teleport(camera.getBukkitEntity().getLocation(), org.bukkit.event.player.PlayerTeleportEvent.TeleportCause.SPECTATE);
+
+                    ServerLevel serverLevel = (ServerLevel) camera.level();
+                    Optional.ofNullable(serverLevel.getChunkSource().chunkMap.entityMap.get(camera.getId())).ifPresent(trackedEntity -> trackedEntity.updatePlayer(this));
+
+                    this.connection.send(new ClientboundSetCameraPacket(camera));
+                } else {
+                    if (!camera.chunkPosition().equals(this.chunkPosition())) {
+                        this.connection.internalTeleport(PositionMoveRotation.of(camera), Collections.emptySet());
+                    }
+
+                    this.absSnapTo(camera.getX(), camera.getY(), camera.getZ(), camera.getYRot(), camera.getXRot());
+                    this.level().getChunkSource().move(this);
+                }
+                // Cheetah end - Fix GLOBAL-1080
                 if (this.wantsToStopRiding()) {
                     this.setCamera(this);
                 }
