--- a/net/minecraft/world/entity/projectile/Projectile.java
+++ b/net/minecraft/world/entity/projectile/Projectile.java
@@ -412,7 +_,10 @@
             if (owner instanceof net.minecraft.server.level.ServerPlayer && target instanceof net.minecraft.server.level.ServerPlayer) {
                 org.bukkit.entity.Player collided = (org.bukkit.entity.Player) target.getBukkitEntity();
                 org.bukkit.entity.Player shooter = (org.bukkit.entity.Player) owner.getBukkitEntity();
-                if (!shooter.canSee(collided)) {
+                // Cheetah start - Added ProjectileAllowCollideWithPlayerEvent
+                net.gommehd.cheetah.event.entity.ProjectileAllowCollideWithPlayerEvent allowCollideEvent = org.bukkit.craftbukkit.event.CraftEventFactory.callProjectileAllowCollideWithPlayerEvent(this, target);
+                if (!allowCollideEvent.isAllowCollide() && !shooter.canSee(collided)) {
+                    // Cheetah end
                     return false;
                 }
             }
