--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -3577,7 +_,17 @@
     }
 
     public boolean isAlliedTo(@Nullable Team team) {
-        return this.getTeam() != null && this.getTeam().isAlliedTo(team);
+        // Cheetah start - Return false when the allied team is the collide rule team
+        final Team ownTeam = this.getTeam();
+        if (ownTeam != null) {
+            if (!io.papermc.paper.configuration.GlobalConfiguration.get().collisions.enablePlayerCollisions &&
+                ownTeam.getName().equals(MinecraftServer.getServer().getPlayerList().collideRuleTeamName)) {
+                return false;
+            }
+            return ownTeam.isAlliedTo(team);
+        }
+        return false;
+        // Cheetah end - Return false when the allied team is the collide rule team
     }
 
     public void setInvisible(boolean invisible) {
