--- a/src/main/java/com/destroystokyo/paper/PaperVersionFetcher.java
+++ b/src/main/java/com/destroystokyo/paper/PaperVersionFetcher.java
@@ -40,8 +_,8 @@
     private static final ComponentLogger COMPONENT_LOGGER = ComponentLogger.logger(LogManager.getRootLogger().getName());
     private static final int DISTANCE_ERROR = -1;
     private static final int DISTANCE_UNKNOWN = -2;
-    private static final String DOWNLOAD_PAGE = "https://papermc.io/downloads/paper";
-    private static final String REPOSITORY = "PaperMC/Paper";
+    private static final String DOWNLOAD_PAGE = "https://github.com/gommehdnet/cheetah"; // Cheetah - Build changes
+    private static final String REPOSITORY = "gommehdnet/cheetah"; // Cheetah - Build changes
     private static final ServerBuildInfo BUILD_INFO;
     private static final String USER_AGENT;
     private static final Gson GSON = new Gson();
@@ -76,54 +_,78 @@
         if (buildNumber.isEmpty() && BUILD_INFO.gitCommit().isEmpty()) {
             COMPONENT_LOGGER.warn(text("*** You are running a development version without access to version information ***"));
         } else {
-            final Optional<MinecraftVersionFetcher> apiResult = fetchMinecraftVersionList();
-            if (buildNumber.isPresent()) {
-                distance = fetchDistanceFromSiteApi(buildNumber.getAsInt());
-            } else {
-                final Optional<String> gitBranch = BUILD_INFO.gitBranch();
-                final Optional<String> gitCommit = BUILD_INFO.gitCommit();
-                if (gitBranch.isPresent() && gitCommit.isPresent()) {
-                    distance = fetchDistanceFromGitHub(gitBranch.get(), gitCommit.get());
-                }
+            // Cheetah start - Build changes
+            final Optional<String> gitBranch = BUILD_INFO.gitBranch();
+            final Optional<String> gitCommit = BUILD_INFO.gitCommit();
+            if (gitBranch.isPresent() && gitCommit.isPresent()) {
+                distance = fetchDistanceFromGitHub(gitBranch.get(), gitCommit.get());
             }
 
             switch (distance) {
                 case DISTANCE_ERROR -> COMPONENT_LOGGER.error(text("*** Error obtaining version information! Cannot fetch version info ***"));
-                case 0 -> apiResult.ifPresent(result -> {
-                    COMPONENT_LOGGER.warn(text("*************************************************************************************"));
-                    COMPONENT_LOGGER.warn(text("You are running the latest build for your Minecraft version (" + BUILD_INFO.minecraftVersionId() + ")"));
-                    COMPONENT_LOGGER.warn(text("However, you are " + result.distance() + " release(s) behind the latest stable release (" + result.latestVersion() + ")!"));
-                    COMPONENT_LOGGER.warn(text("It is recommended that you update as soon as possible"));
-                    COMPONENT_LOGGER.warn(text(DOWNLOAD_PAGE));
-                    COMPONENT_LOGGER.warn(text("*************************************************************************************"));
-                });
+                case 0 -> {}
                 case DISTANCE_UNKNOWN -> COMPONENT_LOGGER.warn(text("*** You are running an unknown version! Cannot fetch version info ***"));
                 default -> {
-                    if (apiResult.isPresent()) {
-                        COMPONENT_LOGGER.warn(text("*** You are running an outdated version of Minecraft, which is " + apiResult.get().distance() + " release(s) and " + distance + " build(s) behind!"));
-                        COMPONENT_LOGGER.warn(text("*** Please update to the latest stable version on " + DOWNLOAD_PAGE + " ***"));
-                    } else {
-                        COMPONENT_LOGGER.info(text("*** Currently you are " + distance + " build(s) behind ***"));
-                        COMPONENT_LOGGER.info(text("*** It is highly recommended to download the latest build from " + DOWNLOAD_PAGE + " ***"));
-                    }
+                    COMPONENT_LOGGER.info(text("*** Currently you are " + distance + " build(s) behind ***"));
+                    COMPONENT_LOGGER.info(text("*** It is highly recommended to download the latest build from " + DOWNLOAD_PAGE + " ***"));
                 }
             }
+            // final Optional<MinecraftVersionFetcher> apiResult = fetchMinecraftVersionList();
+            // if (buildNumber.isPresent()) {
+            //     distance = fetchDistanceFromSiteApi(buildNumber.getAsInt());
+            // } else {
+            //     final Optional<String> gitBranch = BUILD_INFO.gitBranch();
+            //     final Optional<String> gitCommit = BUILD_INFO.gitCommit();
+            //     if (gitBranch.isPresent() && gitCommit.isPresent()) {
+            //         distance = fetchDistanceFromGitHub(gitBranch.get(), gitCommit.get());
+            //     }
+            // }
+            //
+            // switch (distance) {
+            //     case DISTANCE_ERROR -> COMPONENT_LOGGER.error(text("*** Error obtaining version information! Cannot fetch version info ***"));
+            //     case 0 -> apiResult.ifPresent(result -> {
+            //         COMPONENT_LOGGER.warn(text("*************************************************************************************"));
+            //         COMPONENT_LOGGER.warn(text("You are running the latest build for your Minecraft version (" + BUILD_INFO.minecraftVersionId() + ")"));
+            //         COMPONENT_LOGGER.warn(text("However, you are " + result.distance() + " release(s) behind the latest stable release (" + result.latestVersion() + ")!"));
+            //         COMPONENT_LOGGER.warn(text("It is recommended that you update as soon as possible"));
+            //         COMPONENT_LOGGER.warn(text(DOWNLOAD_PAGE));
+            //         COMPONENT_LOGGER.warn(text("*************************************************************************************"));
+            //     });
+            //     case DISTANCE_UNKNOWN -> COMPONENT_LOGGER.warn(text("*** You are running an unknown version! Cannot fetch version info ***"));
+            //     default -> {
+            //         if (apiResult.isPresent()) {
+            //             COMPONENT_LOGGER.warn(text("*** You are running an outdated version of Minecraft, which is " + apiResult.get().distance() + " release(s) and " + distance + " build(s) behind!"));
+            //             COMPONENT_LOGGER.warn(text("*** Please update to the latest stable version on " + DOWNLOAD_PAGE + " ***"));
+            //         } else {
+            //             COMPONENT_LOGGER.info(text("*** Currently you are " + distance + " build(s) behind ***"));
+            //             COMPONENT_LOGGER.info(text("*** It is highly recommended to download the latest build from " + DOWNLOAD_PAGE + " ***"));
+            //         }
+            //     }
+            // }
+            // Cheetah end - Build changes
         }
     }
 
     private static Component getUpdateStatusMessage() {
         int distance = DISTANCE_ERROR;
 
-        final OptionalInt buildNumber = PaperVersionFetcher.BUILD_INFO.buildNumber();
-        if (buildNumber.isPresent()) {
-            distance = fetchDistanceFromSiteApi(buildNumber.getAsInt());
-        } else {
-            final Optional<String> gitBranch = PaperVersionFetcher.BUILD_INFO.gitBranch();
-            final Optional<String> gitCommit = PaperVersionFetcher.BUILD_INFO.gitCommit();
-            if (gitBranch.isPresent() && gitCommit.isPresent()) {
-                distance = fetchDistanceFromGitHub(gitBranch.get(), gitCommit.get());
-            }
+        // Cheetah start - Build changes
+        final Optional<String> gitBranch = PaperVersionFetcher.BUILD_INFO.gitBranch();
+        final Optional<String> gitCommit = PaperVersionFetcher.BUILD_INFO.gitCommit();
+        if (gitBranch.isPresent() && gitCommit.isPresent()) {
+            distance = fetchDistanceFromGitHub(gitBranch.get(), gitCommit.get());
         }
+        // final OptionalInt buildNumber = PaperVersionFetcher.BUILD_INFO.buildNumber();
+        // if (buildNumber.isPresent()) {
+        //     distance = fetchDistanceFromSiteApi(buildNumber.getAsInt());
+        // } else {
+        //     final Optional<String> gitBranch = PaperVersionFetcher.BUILD_INFO.gitBranch();
+        //     final Optional<String> gitCommit = PaperVersionFetcher.BUILD_INFO.gitCommit();
+        //     if (gitBranch.isPresent() && gitCommit.isPresent()) {
+        //         distance = fetchDistanceFromGitHub(gitBranch.get(), gitCommit.get());
+        //     }
+        // }
+        // Cheetah end - Build changes
 
         return switch (distance) {
             case DISTANCE_ERROR -> text("Error obtaining version information", NamedTextColor.YELLOW);
