--- a/net/minecraft/network/protocol/game/ClientboundAnimatePacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundAnimatePacket.java
@@ -23,6 +_,13 @@
         this.action = action;
     }
 
+    // Cheetah start - Expose constructor with entity id
+    public ClientboundAnimatePacket(int id, int action) {
+        this.id = id;
+        this.action = action;
+    }
+    // Cheetah end - Expose constructor with entity id
+
     private ClientboundAnimatePacket(final FriendlyByteBuf input) {
         this.id = input.readVarInt();
         this.action = input.readUnsignedByte();
