--- a/net/minecraft/network/protocol/game/ClientboundLevelChunkPacketData.java
+++ b/net/minecraft/network/protocol/game/ClientboundLevelChunkPacketData.java
@@ -153,7 +_,13 @@
         return this.heightmaps;
     }
 
-    static class BlockEntityInfo {
+    // Cheetah start - add getter
+    public List<ClientboundLevelChunkPacketData.BlockEntityInfo> getBlockEntitiesData() {
+        return blockEntitiesData;
+    }
+    // Cheetah end - add getter
+
+    public static class BlockEntityInfo {
         public static final StreamCodec<RegistryFriendlyByteBuf, ClientboundLevelChunkPacketData.BlockEntityInfo> STREAM_CODEC = StreamCodec.ofMember(
             ClientboundLevelChunkPacketData.BlockEntityInfo::write, ClientboundLevelChunkPacketData.BlockEntityInfo::new
         );
@@ -186,6 +_,25 @@
             ByteBufCodecs.registry(Registries.BLOCK_ENTITY_TYPE).encode(buffer, this.type);
             buffer.writeNbt(this.tag);
         }
+
+        // Cheetah start - add getter
+        public int getPackedXZ() {
+            return packedXZ;
+        }
+
+        public int getY() {
+            return y;
+        }
+
+        public BlockEntityType<?> getType() {
+            return type;
+        }
+
+        @Nullable
+        public CompoundTag getTag() {
+            return tag;
+        }
+        // Cheetah end - add getter
 
         static ClientboundLevelChunkPacketData.BlockEntityInfo create(BlockEntity blockEntity) {
             CompoundTag updateTag = blockEntity.getUpdateTag(blockEntity.getLevel().registryAccess());
