--- a/net/minecraft/network/protocol/game/ClientboundSetCameraPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundSetCameraPacket.java
@@ -18,6 +_,12 @@
         this.cameraId = cameraEntity.getId();
     }
 
+    // Cheetah start - add constructor with entity id
+    public ClientboundSetCameraPacket(int cameraId) {
+        this.cameraId = cameraId;
+    }
+    // Cheetah end - add constructor with entity id
+
     private ClientboundSetCameraPacket(FriendlyByteBuf buffer) {
         this.cameraId = buffer.readVarInt();
     }
