--- a/net/minecraft/network/protocol/game/ClientboundSetCameraPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundSetCameraPacket.java
@@ -18,6 +_,12 @@
         this.cameraId = camera.getId();
     }
 
+    // Cheetah start - Expose constructor with entity id
+    public ClientboundSetCameraPacket(int cameraId) {
+        this.cameraId = cameraId;
+    }
+    // Cheetah end - Expose constructor with entity id
+
     private ClientboundSetCameraPacket(final FriendlyByteBuf input) {
         this.cameraId = input.readVarInt();
     }
