--- a/net/minecraft/network/protocol/game/ClientboundRotateHeadPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundRotateHeadPacket.java
@@ -21,6 +_,13 @@
         this.yHeadRot = yHeadRot;
     }
 
+    // Cheetah start - Expose constructor with entity id
+    public ClientboundRotateHeadPacket(int entityId, byte yHeadRot) {
+        this.entityId = entityId;
+        this.yHeadRot = yHeadRot;
+    }
+    // Cheetah end - Expose constructor with entity id
+
     private ClientboundRotateHeadPacket(final FriendlyByteBuf input) {
         this.entityId = input.readVarInt();
         this.yHeadRot = input.readByte();
