--- a/net/minecraft/network/protocol/game/ClientboundEntityEventPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundEntityEventPacket.java
@@ -20,6 +_,13 @@
         this.eventId = eventId;
     }
 
+    // Cheetah start - Expose constructor with entity id
+    public ClientboundEntityEventPacket(int entityId, byte eventId) {
+        this.entityId = entityId;
+        this.eventId = eventId;
+    }
+    // Cheetah end - Expose constructor with entity id
+
     private ClientboundEntityEventPacket(final FriendlyByteBuf input) {
         this.entityId = input.readInt();
         this.eventId = input.readByte();
