--- a/net/minecraft/network/protocol/game/ClientboundLevelChunkPacketData.java
+++ b/net/minecraft/network/protocol/game/ClientboundLevelChunkPacketData.java
@@ -153,6 +_,12 @@
         return this.heightmaps;
     }
 
+    // Cheetah start - Expose block entities data
+    public List<ClientboundLevelChunkPacketData.BlockEntityInfo> getBlockEntitiesData() {
+        return this.blockEntitiesData;
+    }
+    // Cheetah end - Expose block entities data
+
     public static class BlockEntityInfo {
         public static final StreamCodec<RegistryFriendlyByteBuf, ClientboundLevelChunkPacketData.BlockEntityInfo> STREAM_CODEC = StreamCodec.ofMember(
             ClientboundLevelChunkPacketData.BlockEntityInfo::write, ClientboundLevelChunkPacketData.BlockEntityInfo::new
@@ -185,6 +_,25 @@
             ByteBufCodecs.registry(Registries.BLOCK_ENTITY_TYPE).encode(output, this.type);
             output.writeNbt(this.tag);
         }
+
+        // Cheetah start - Expose getter
+        public int getPackedXZ() {
+            return this.packedXZ;
+        }
+
+        public int getY() {
+            return this.y;
+        }
+
+        public BlockEntityType<?> getType() {
+            return this.type;
+        }
+
+        @Nullable
+        public CompoundTag getTag() {
+            return this.tag;
+        }
+        // Cheetah end - Expose getter
 
         private static ClientboundLevelChunkPacketData.BlockEntityInfo create(final BlockEntity blockEntity) {
             CompoundTag tag = blockEntity.getUpdateTag(blockEntity.getLevel().registryAccess());
