--- a/net/minecraft/network/protocol/game/ClientboundRotateHeadPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundRotateHeadPacket.java
@@ -21,6 +_,13 @@
         this.yHeadRot = yHeadRot;
     }
 
+    // Cheetah start - add constructor with entity id
+    public ClientboundRotateHeadPacket(int entityId, byte yHeadRot) {
+        this.entityId = entityId;
+        this.yHeadRot = yHeadRot;
+    }
+    // Cheetah end - add constructor with entity id
+
     private ClientboundRotateHeadPacket(FriendlyByteBuf buffer) {
         this.entityId = buffer.readVarInt();
         this.yHeadRot = buffer.readByte();
