--- a/net/minecraft/network/protocol/game/ServerboundInteractPacket.java
+++ b/net/minecraft/network/protocol/game/ServerboundInteractPacket.java
@@ -162,6 +_,12 @@
     }
     // Paper end - PlayerUseUnknownEntityEvent
 
+    // Cheetah start - Expose action getter
+    public ServerboundInteractPacket.Action getAction() {
+        return this.action;
+    }
+    // Cheetah end - Expose action getter
+
     static class InteractionAtLocationAction implements ServerboundInteractPacket.Action {
         private final InteractionHand hand;
         private final Vec3 location;
