From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: lukas81298 <lukas81298@gommehd.net>
Date: Sun, 10 Jul 2022 22:50:44 +0200
Subject: [PATCH] API for configurable hopper transfer rates


diff --git a/src/main/java/org/bukkit/block/Hopper.java b/src/main/java/org/bukkit/block/Hopper.java
index 61ea33c1f2dbb546a66f945a01feae437b1381e0..ee7f2b5dff0de861568748efb477ced5a7b0bd71 100644
--- a/src/main/java/org/bukkit/block/Hopper.java
+++ b/src/main/java/org/bukkit/block/Hopper.java
@@ -21,5 +21,30 @@ public interface Hopper extends Container, LootableBlockInventory { // Paper
      */
     int getTransferCooldown();
     // Paper end - Expanded Hopper API
+
+    // Cheetah start - API for configurable hopper transfer rates
+    /**
+     * Returns the current transfer multiplier for this hopper
+     * Default value: 1.0
+     * If set to 2.0, twice as many items will be transferred per time unit
+     *
+     * @return transfer multiplier for this hopper
+     */
+    float getTransferMultiplier();
+
+    /**
+     * Sets the transfer multiplier for this hopper. The value is stored persistently.
+     * Default value: 1.0
+     * If set to 2.0, twice as many items will be transferred per time unit
+     */
+    void setTransferMultiplier(float transferMultiplier);
+
+    /**
+     * Resets the transfer multiplier to the default value
+     */
+    default void resetTransferMultiplier() {
+        this.setTransferMultiplier(0F);
+    }
+    // Cheetah end
 }
 
