From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Summerfeeling <summerfeelingmc@gmail.com>
Date: Wed, 11 Jan 2023 14:11:21 +0100
Subject: [PATCH] Async-able StructuresLocateEvent


diff --git a/src/main/java/io/papermc/paper/event/world/StructuresLocateEvent.java b/src/main/java/io/papermc/paper/event/world/StructuresLocateEvent.java
index d39b3dc48079fc83f1fd8e7ecde0d4ae77b635ce..3bacbf6f62f6e59cb39f672c8a474b31846a0832 100644
--- a/src/main/java/io/papermc/paper/event/world/StructuresLocateEvent.java
+++ b/src/main/java/io/papermc/paper/event/world/StructuresLocateEvent.java
@@ -50,6 +50,16 @@ public class StructuresLocateEvent extends WorldEvent implements Cancellable {
         this.findUnexplored = findUnexplored;
     }
 
+    // Cheetah start - added async parameter
+    public StructuresLocateEvent(final World world, final Location origin, final List<Structure> structures, final int radius, final boolean findUnexplored, final boolean async) {
+        super(world, async);
+        this.origin = origin;
+        this.setStructures(structures);
+        this.radius = radius;
+        this.findUnexplored = findUnexplored;
+    }
+    // Cheetah end
+
     /**
      * Gets the {@link Location} from which the search is to be conducted.
      *
