From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Summerfeeling <summerfeelingmc@gmail.com>
Date: Wed, 11 Jan 2023 14:11:21 +0100
Subject: [PATCH] Async-able StructuresLocateEvent


diff --git a/src/main/java/io/papermc/paper/event/world/StructuresLocateEvent.java b/src/main/java/io/papermc/paper/event/world/StructuresLocateEvent.java
index 4342b162f69b6509503e9772ddd2c5ac6659545a..6f913cf5e617c7cb65e7343e6f8e5d9e98d4867b 100644
--- a/src/main/java/io/papermc/paper/event/world/StructuresLocateEvent.java
+++ b/src/main/java/io/papermc/paper/event/world/StructuresLocateEvent.java
@@ -50,6 +50,14 @@ public class StructuresLocateEvent extends WorldEvent implements Cancellable {
         this.findUnexplored = findUnexplored;
     }
 
+    public StructuresLocateEvent(@NotNull World world, @NotNull Location origin, @NotNull List<ConfiguredStructure> configuredStructures, int radius, boolean findUnexplored, boolean async) {
+        super(world, async);
+        this.origin = origin;
+        this.configuredStructures = configuredStructures;
+        this.radius = radius;
+        this.findUnexplored = findUnexplored;
+    }
+
     /**
      * Gets the {@link Location} from which the search is to be conducted.
      *
