From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Summerfeeling <summerfeelingmc@gmail.com>
Date: Wed, 11 Jan 2023 14:11:21 +0100
Subject: [PATCH] Async-able StructuresLocateEvent


diff --git a/src/main/java/io/papermc/paper/event/world/StructuresLocateEvent.java b/src/main/java/io/papermc/paper/event/world/StructuresLocateEvent.java
index 41ea65c9ecf6339bb50864a6d28e53c7e8d1edf7..79f094559c4908d6f204568af7411cc3bb4e4d3e 100644
--- a/src/main/java/io/papermc/paper/event/world/StructuresLocateEvent.java
+++ b/src/main/java/io/papermc/paper/event/world/StructuresLocateEvent.java
@@ -49,6 +49,16 @@ public class StructuresLocateEvent extends WorldEvent implements Cancellable {
         this.findUnexplored = findUnexplored;
     }
 
+    // Cheetah start - added async parameter
+    public StructuresLocateEvent(@NotNull World world, @NotNull Location origin, @NotNull List<Structure> structures, int radius, boolean findUnexplored, boolean async) {
+        super(world, async);
+        this.origin = origin;
+        this.setStructures(structures);
+        this.radius = radius;
+        this.findUnexplored = findUnexplored;
+    }
+    // Cheetah end
+
     /**
      * Gets the {@link Location} from which the search is to be conducted.
      *
