From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: lukas81298 <lukas81298@gommehd.net>
Date: Mon, 13 Mar 2023 18:35:28 +0100
Subject: [PATCH] updated upstream


diff --git a/src/main/java/io/papermc/paper/event/world/StructuresLocateEvent.java b/src/main/java/io/papermc/paper/event/world/StructuresLocateEvent.java
index 5cc563936e7f2370c248f09672b00b6f524b5aa2..8faec65a3b27c2c9376860add0e90e8742f7cffa 100644
--- a/src/main/java/io/papermc/paper/event/world/StructuresLocateEvent.java
+++ b/src/main/java/io/papermc/paper/event/world/StructuresLocateEvent.java
@@ -53,13 +53,15 @@ public class StructuresLocateEvent extends WorldEvent implements Cancellable {
         this.findUnexplored = findUnexplored;
     }
 
-    public StructuresLocateEvent(@NotNull World world, @NotNull Location origin, @NotNull List<ConfiguredStructure> configuredStructures, int radius, boolean findUnexplored, boolean async) {
+    // Cheetah start - added async parameter
+    public StructuresLocateEvent(@NotNull World world, @NotNull Location origin, @NotNull List<Structure> structures, int radius, boolean findUnexplored, boolean async) {
         super(world, async);
         this.origin = origin;
-        this.configuredStructures = configuredStructures;
+        this.setStructures(structures);
         this.radius = radius;
         this.findUnexplored = findUnexplored;
     }
+    // Cheetah end
 
     /**
      * Gets the {@link Location} from which the search is to be conducted.
