From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: lukas81298 <lukas81298@gommehd.net>
Date: Wed, 5 Apr 2023 22:55:11 +0200
Subject: [PATCH] allow skipping log messages for canceled login events


diff --git a/src/main/java/org/bukkit/event/player/AsyncPlayerPreLoginEvent.java b/src/main/java/org/bukkit/event/player/AsyncPlayerPreLoginEvent.java
index ae02788536d0c6736aba4ccf812844df4f9789f1..7d47b0a324536a70ee6e7ed1235a2499e69ae868 100644
--- a/src/main/java/org/bukkit/event/player/AsyncPlayerPreLoginEvent.java
+++ b/src/main/java/org/bukkit/event/player/AsyncPlayerPreLoginEvent.java
@@ -27,6 +27,7 @@ public class AsyncPlayerPreLoginEvent extends Event {
     private final InetAddress rawAddress; // Paper
     private final String hostname; // Paper
     private final boolean transferred;
+    private boolean skipLog; // Cheetah - Allow skipping log messages for canceled login events
 
     @Deprecated
     public AsyncPlayerPreLoginEvent(@NotNull final String name, @NotNull final InetAddress ipAddress) {
@@ -297,6 +298,16 @@ public class AsyncPlayerPreLoginEvent extends Event {
         return transferred;
     }
 
+    // Cheetah start - Allow skipping log messages for canceled login events
+    public boolean isSkipLog() {
+        return this.skipLog;
+    }
+
+    public void setSkipLog(boolean skipLog) {
+        this.skipLog = skipLog;
+    }
+    // Cheetah end
+
     @NotNull
     @Override
     public HandlerList getHandlers() {
diff --git a/src/main/java/org/bukkit/event/player/PlayerLoginEvent.java b/src/main/java/org/bukkit/event/player/PlayerLoginEvent.java
index eaa0548cf430bf5b58ff84e0a4403c451699db28..2a07caef10e00123818ac288a1cbccb894fa781a 100644
--- a/src/main/java/org/bukkit/event/player/PlayerLoginEvent.java
+++ b/src/main/java/org/bukkit/event/player/PlayerLoginEvent.java
@@ -19,6 +19,7 @@ public class PlayerLoginEvent extends PlayerEvent {
     private final String hostname;
     private Result result = Result.ALLOWED;
     private net.kyori.adventure.text.Component message = net.kyori.adventure.text.Component.empty();
+    private boolean skipLog; // Cheetah - Allow skipping log messages for canceled login events
 
     /**
      * This constructor defaults message to an empty string, and result to
@@ -161,6 +162,16 @@ public class PlayerLoginEvent extends PlayerEvent {
         return hostname;
     }
 
+    // Cheetah start - Allow skipping log messages for canceled login events
+    public boolean isSkipLog() {
+        return this.skipLog;
+    }
+
+    public void setSkipLog(boolean skipLog) {
+        this.skipLog = skipLog;
+    }
+    // Cheetah end
+
     /**
      * Allows the player to log in
      */
diff --git a/src/main/java/org/bukkit/event/player/PlayerPreLoginEvent.java b/src/main/java/org/bukkit/event/player/PlayerPreLoginEvent.java
index 175ed12dd1698f4d153c9acdac8340c15a427ea5..36103261a69bf8c7da83c19e0ce462a572d7df98 100644
--- a/src/main/java/org/bukkit/event/player/PlayerPreLoginEvent.java
+++ b/src/main/java/org/bukkit/event/player/PlayerPreLoginEvent.java
@@ -30,6 +30,7 @@ public class PlayerPreLoginEvent extends Event {
     private final String name;
     private final InetAddress ipAddress;
     private final UUID uniqueId;
+    private boolean skipLog; // Cheetah - Allow skipping log messages for canceled login events
 
     @Deprecated
     public PlayerPreLoginEvent(@NotNull final String name, @NotNull final InetAddress ipAddress) {
@@ -176,6 +177,16 @@ public class PlayerPreLoginEvent extends Event {
         return uniqueId;
     }
 
+    // Cheetah start - Allow skipping log messages for canceled login events
+    public boolean isSkipLog() {
+        return this.skipLog;
+    }
+
+    public void setSkipLog(boolean skipLog) {
+        this.skipLog = skipLog;
+    }
+    // Cheetah end
+
     @NotNull
     public static HandlerList getHandlerList() {
         return handlers;
