From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: lukas81298 <lukas81298@gommehd.net>
Date: Wed, 5 Apr 2023 22:55:11 +0200
Subject: [PATCH] allow skipping log messages for canceled login events


diff --git a/src/main/java/org/bukkit/event/player/AsyncPlayerPreLoginEvent.java b/src/main/java/org/bukkit/event/player/AsyncPlayerPreLoginEvent.java
index bdb7d2345e7d3a48a2f0561832ce126a16781e89..07ea92f5b658174429fe7f070dcce828f3b7d183 100644
--- a/src/main/java/org/bukkit/event/player/AsyncPlayerPreLoginEvent.java
+++ b/src/main/java/org/bukkit/event/player/AsyncPlayerPreLoginEvent.java
@@ -28,6 +28,7 @@ public class AsyncPlayerPreLoginEvent extends Event {
     private PlayerProfile profile; // Paper
     private final InetAddress rawAddress; // Paper
     private final String hostname; // Paper
+    private boolean skipLog; // Cheetah - Allow skipping log messages for canceled login events
 
     @Deprecated
     public AsyncPlayerPreLoginEvent(@NotNull final String name, @NotNull final InetAddress ipAddress) {
@@ -282,6 +283,16 @@ public class AsyncPlayerPreLoginEvent extends Event {
     }
     // Paper end
 
+    // Cheetah start - Allow skipping log messages for canceled login events
+    public boolean isSkipLog() {
+        return skipLog;
+    }
+
+    public void setSkipLog(boolean skipLog) {
+        this.skipLog = skipLog;
+    }
+    // Cheetah end
+
     @NotNull
     @Override
     public HandlerList getHandlers() {
diff --git a/src/main/java/org/bukkit/event/player/PlayerLoginEvent.java b/src/main/java/org/bukkit/event/player/PlayerLoginEvent.java
index eaa0548cf430bf5b58ff84e0a4403c451699db28..d8e8d015a9046a1171b009befd1f6d4a56fb5e7d 100644
--- a/src/main/java/org/bukkit/event/player/PlayerLoginEvent.java
+++ b/src/main/java/org/bukkit/event/player/PlayerLoginEvent.java
@@ -19,6 +19,7 @@ public class PlayerLoginEvent extends PlayerEvent {
     private final String hostname;
     private Result result = Result.ALLOWED;
     private net.kyori.adventure.text.Component message = net.kyori.adventure.text.Component.empty();
+    private boolean skipLog; // Cheetah - Allow skipping log messages for canceled login events
 
     /**
      * This constructor defaults message to an empty string, and result to
@@ -161,6 +162,16 @@ public class PlayerLoginEvent extends PlayerEvent {
         return hostname;
     }
 
+    // Cheetah start - Allow skipping log messages for canceled login events
+    public boolean isSkipLog() {
+        return skipLog;
+    }
+
+    public void setSkipLog(boolean skipLog) {
+        this.skipLog = skipLog;
+    }
+    // Cheetah end
+
     /**
      * Allows the player to log in
      */
diff --git a/src/main/java/org/bukkit/event/player/PlayerPreLoginEvent.java b/src/main/java/org/bukkit/event/player/PlayerPreLoginEvent.java
index 175ed12dd1698f4d153c9acdac8340c15a427ea5..492b32f94d2077e5c7ba714873b315f49c9a0c12 100644
--- a/src/main/java/org/bukkit/event/player/PlayerPreLoginEvent.java
+++ b/src/main/java/org/bukkit/event/player/PlayerPreLoginEvent.java
@@ -30,6 +30,7 @@ public class PlayerPreLoginEvent extends Event {
     private final String name;
     private final InetAddress ipAddress;
     private final UUID uniqueId;
+    private boolean skipLog; // Cheetah - Allow skipping log messages for canceled login events
 
     @Deprecated
     public PlayerPreLoginEvent(@NotNull final String name, @NotNull final InetAddress ipAddress) {
@@ -176,6 +177,16 @@ public class PlayerPreLoginEvent extends Event {
         return uniqueId;
     }
 
+    // Cheetah start - Allow skipping log messages for canceled login events
+    public boolean isSkipLog() {
+        return skipLog;
+    }
+
+    public void setSkipLog(boolean skipLog) {
+        this.skipLog = skipLog;
+    }
+    // Cheetah end
+
     @NotNull
     public static HandlerList getHandlerList() {
         return handlers;
