From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: lukas81298 <lukas81298@gommehd.net>
Date: Mon, 13 Mar 2023 18:35:28 +0100
Subject: [PATCH] updated upstream


diff --git a/src/main/java/io/papermc/paper/event/world/StructuresLocateEvent.java b/src/main/java/io/papermc/paper/event/world/StructuresLocateEvent.java
index 6f913cf5e617c7cb65e7343e6f8e5d9e98d4867b..aaf618d506c1042004d18bc7776c7c45a6d5bdfe 100644
--- a/src/main/java/io/papermc/paper/event/world/StructuresLocateEvent.java
+++ b/src/main/java/io/papermc/paper/event/world/StructuresLocateEvent.java
@@ -50,13 +50,15 @@ public class StructuresLocateEvent extends WorldEvent implements Cancellable {
         this.findUnexplored = findUnexplored;
     }
 
-    public StructuresLocateEvent(@NotNull World world, @NotNull Location origin, @NotNull List<ConfiguredStructure> configuredStructures, int radius, boolean findUnexplored, boolean async) {
+    // Cheetah start - added async parameter
+    public StructuresLocateEvent(@NotNull World world, @NotNull Location origin, @NotNull List<Structure> structures, int radius, boolean findUnexplored, boolean async) {
         super(world, async);
         this.origin = origin;
-        this.configuredStructures = configuredStructures;
+        this.setStructures(structures);
         this.radius = radius;
         this.findUnexplored = findUnexplored;
     }
+    // Cheetah end
 
     /**
      * Gets the {@link Location} from which the search is to be conducted.
