From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: lukas81298 <lukas81298@gommehd.net>
Date: Wed, 5 Apr 2023 22:55:11 +0200
Subject: [PATCH] allow skipping log messages for canceled login events


diff --git a/src/main/java/org/bukkit/event/player/AsyncPlayerPreLoginEvent.java b/src/main/java/org/bukkit/event/player/AsyncPlayerPreLoginEvent.java
index a74c88ef229b0e4d359978fd79b23b56141e7146..9037df22bda1b8eea7803bde35133f0a95914d82 100644
--- a/src/main/java/org/bukkit/event/player/AsyncPlayerPreLoginEvent.java
+++ b/src/main/java/org/bukkit/event/player/AsyncPlayerPreLoginEvent.java
@@ -30,6 +30,7 @@ public class AsyncPlayerPreLoginEvent extends Event {
     private final InetAddress rawAddress; // Paper
     //private UUID uniqueId; // Paper - Not used anymore
     private final String hostname; // Paper
+    private boolean skipLog;
 
     @Deprecated
     public AsyncPlayerPreLoginEvent(@NotNull final String name, @NotNull final InetAddress ipAddress) {
@@ -291,6 +292,15 @@ public class AsyncPlayerPreLoginEvent extends Event {
     }
     // Paper end
 
+
+    public boolean isSkipLog() {
+        return skipLog;
+    }
+
+    public void setSkipLog(boolean skipLog) {
+        this.skipLog = skipLog;
+    }
+
     @NotNull
     @Override
     public HandlerList getHandlers() {
diff --git a/src/main/java/org/bukkit/event/player/PlayerLoginEvent.java b/src/main/java/org/bukkit/event/player/PlayerLoginEvent.java
index eaa0548cf430bf5b58ff84e0a4403c451699db28..953d6fdfd232fc3c9176c4f19601134b0241edf4 100644
--- a/src/main/java/org/bukkit/event/player/PlayerLoginEvent.java
+++ b/src/main/java/org/bukkit/event/player/PlayerLoginEvent.java
@@ -19,6 +19,7 @@ public class PlayerLoginEvent extends PlayerEvent {
     private final String hostname;
     private Result result = Result.ALLOWED;
     private net.kyori.adventure.text.Component message = net.kyori.adventure.text.Component.empty();
+    private boolean skipLog;
 
     /**
      * This constructor defaults message to an empty string, and result to
@@ -161,6 +162,14 @@ public class PlayerLoginEvent extends PlayerEvent {
         return hostname;
     }
 
+    public boolean isSkipLog() {
+        return skipLog;
+    }
+
+    public void setSkipLog(boolean skipLog) {
+        this.skipLog = skipLog;
+    }
+
     /**
      * Allows the player to log in
      */
diff --git a/src/main/java/org/bukkit/event/player/PlayerPreLoginEvent.java b/src/main/java/org/bukkit/event/player/PlayerPreLoginEvent.java
index 175ed12dd1698f4d153c9acdac8340c15a427ea5..b9983f610ab86d27bc72ee90cc6747ea891f5d19 100644
--- a/src/main/java/org/bukkit/event/player/PlayerPreLoginEvent.java
+++ b/src/main/java/org/bukkit/event/player/PlayerPreLoginEvent.java
@@ -30,6 +30,7 @@ public class PlayerPreLoginEvent extends Event {
     private final String name;
     private final InetAddress ipAddress;
     private final UUID uniqueId;
+    private boolean skipLog;
 
     @Deprecated
     public PlayerPreLoginEvent(@NotNull final String name, @NotNull final InetAddress ipAddress) {
@@ -176,6 +177,14 @@ public class PlayerPreLoginEvent extends Event {
         return uniqueId;
     }
 
+    public boolean isSkipLog() {
+        return skipLog;
+    }
+
+    public void setSkipLog(boolean skipLog) {
+        this.skipLog = skipLog;
+    }
+
     @NotNull
     public static HandlerList getHandlerList() {
         return handlers;
