From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jendrik Eggers <jendrikeggerskapp@web.de>
Date: Sat, 9 Dec 2023 03:37:57 +0100
Subject: [PATCH] ignore javadoc errors


diff --git a/build.gradle.kts b/build.gradle.kts
index 571534b42cd9c33d6a7bb6fe3bf3a28e33f8e5de..8daeb74ae931c1400971912bce5f4623f643c078 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -162,6 +162,7 @@ tasks.jar {
 }
 
 tasks.withType<Javadoc> {
+    (options as StandardJavadocDocletOptions).addStringOption("Xdoclint:none", "-quiet") // Cheetah - ignore javadoc errors
     val options = options as StandardJavadocDocletOptions
     options.overview = "src/main/javadoc/overview.html"
     options.use()
