From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jendrik Eggers <jendrikeggerskapp@web.de>
Date: Sat, 9 Dec 2023 03:37:57 +0100
Subject: [PATCH] ignore javadoc errors


diff --git a/build.gradle.kts b/build.gradle.kts
index e827ee211e3c65dc68ac5867fd8476639df63645..ba2cdcdd2cbf8bfa1421a6bf93686d222a6826a2 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -118,6 +118,7 @@ tasks.jar {
 }
 
 tasks.withType<Javadoc> {
+    (options as StandardJavadocDocletOptions).addStringOption("Xdoclint:none", "-quiet") // Cheetah - ignore javadoc errors
     val options = options as StandardJavadocDocletOptions
     options.overview = "src/main/javadoc/overview.html"
     options.use()
