--- a/net/minecraft/server/network/ServerLoginPacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerLoginPacketListenerImpl.java
@@ -71,6 +_,7 @@
     private int velocityLoginMessageId = -1; // Paper - Add Velocity IP Forwarding Support
     public java.util.@Nullable UUID requestedUuid; // Paper
     private final io.papermc.paper.connection.PaperPlayerLoginConnection paperLoginConnection; // Paper - Config API
+    private boolean disconnectLogged = false; // Cheetah - Allow skipping log messages for canceled login events
 
     public ServerLoginPacketListenerImpl(final MinecraftServer minecraftserver, final Connection connection, final boolean transferred) {
         this.server = minecraftserver;
@@ -124,8 +_,15 @@
 
     // Paper start
     public void disconnectAsync(Component reason) {
+        disconnectAsync(reason, false);
+    }
+
+    public void disconnectAsync(Component reason, boolean skipLog) {
         try {
-            LOGGER.info("Disconnecting {}: {}", this.getUserName(), reason.getString());
+            this.disconnectLogged = true;
+            if (!skipLog) {
+                LOGGER.info("Disconnecting {}: {}", this.getUserName(), reason.getString());
+            }
             this.connection.send(new ClientboundLoginDisconnectPacket(reason), PacketSendListener.thenRun(() -> this.connection.disconnect(reason)));
             this.connection.handleConnectionDisconnectOnNextTick = true;
         } catch (Exception var3) {
@@ -134,8 +_,17 @@
     }
     // Paper end
     public void disconnect(final Component component) {
+        // Cheetah start - Allow skipping log messages for canceled login events
+       disconnect(component, false);
+    }
+
+    public void disconnect(Component component, boolean skipLog) {
         try {
-            LOGGER.info("Disconnecting {}: {}", this.getUserName(), component.getString());
+            this.disconnectLogged = true;
+            if (!skipLog) {
+                LOGGER.info("Disconnecting {}: {}", this.getUserName(), component.getString());
+            }
+            // Cheetah end - Allow skipping log messages for canceled login events
             this.connection.send(new ClientboundLoginDisconnectPacket(component));
             this.connection.disconnect(component);
         } catch (Exception e) {
@@ -149,7 +_,7 @@
 
     @Override
     public void onDisconnect(final DisconnectionDetails details) {
-        LOGGER.info("{} lost connection: {}", this.getUserName(), details.reason().getString());
+        if (!this.disconnectLogged) LOGGER.info("{} lost connection: {}", this.getUserName(), details.reason().getString()); // Cheetah - Allow skipping log messages for canceled login events
     }
 
     public String getUserName() {
@@ -212,9 +_,10 @@
 
     private void verifyLoginAndFinishConnectionSetup(final GameProfile profile) {
         PlayerList playerList = this.server.getPlayerList();
-        Component error = org.bukkit.craftbukkit.event.CraftEventFactory.handleLoginResult(playerList.canPlayerLogin(this.connection.getRemoteAddress(), new NameAndId(profile)), this.paperLoginConnection, this.connection, profile, this.server, true); // Paper
+        io.papermc.paper.connection.HorriblePlayerLoginEventHack.LoginResult loginResult = org.bukkit.craftbukkit.event.CraftEventFactory.handleLoginResult(playerList.canPlayerLogin(this.connection.getRemoteAddress(), new NameAndId(profile)), this.paperLoginConnection, this.connection, profile, this.server, true); // Paper // Cheetah - Allow skipping log messages for canceled login events
+        Component error = loginResult.message(); // Cheetah - Allow skipping log messages for canceled login events
         if (error != null) {
-            this.disconnectAsync(error); // Paper
+            this.disconnectAsync(error, loginResult.isSkipLog()); // Paper // Cheetah - Allow skipping log messages for canceled login events
         } else {
             if (this.server.getCompressionThreshold() >= 0 && !this.connection.isMemoryConnection()) {
                 this.connection
@@ -337,12 +_,14 @@
         playerName = gameprofile.name();
         uniqueId = gameprofile.id();
         // Paper end - Add more fields to AsyncPlayerPreLoginEvent
+        boolean skipLog = asyncEvent.isSkipLog(); // Cheetah - Allow skipping log messages for canceled login events
 
         if (PlayerPreLoginEvent.getHandlerList().getRegisteredListeners().length != 0) {
             final PlayerPreLoginEvent event = new PlayerPreLoginEvent(playerName, address, uniqueId);
             if (asyncEvent.getResult() != PlayerPreLoginEvent.Result.ALLOWED) {
                 event.disallow(asyncEvent.getResult(), asyncEvent.kickMessage()); // Paper - Adventure
             }
+            event.setSkipLog(skipLog); // Cheetah - Allow skipping log messages for canceled login events
             Waitable<PlayerPreLoginEvent.Result> waitable = new Waitable<>() {
                 @Override
                 protected PlayerPreLoginEvent.Result evaluate() {
@@ -353,11 +_,12 @@
 
             ServerLoginPacketListenerImpl.this.server.processQueue.add(waitable);
             if (waitable.get() != PlayerPreLoginEvent.Result.ALLOWED) {
-                this.disconnect(io.papermc.paper.adventure.PaperAdventure.asVanilla(event.kickMessage())); // Paper - Adventure
+                skipLog = event.isSkipLog(); // Cheetah - Allow skipping log messages for canceled login events
+                this.disconnect(io.papermc.paper.adventure.PaperAdventure.asVanilla(event.kickMessage()), skipLog); // Paper - Adventure // Cheetah - Allow skipping log messages for canceled login events
             }
         } else {
             if (asyncEvent.getLoginResult() != AsyncPlayerPreLoginEvent.Result.ALLOWED) {
-                this.disconnect(io.papermc.paper.adventure.PaperAdventure.asVanilla(asyncEvent.kickMessage())); // Paper - Adventure
+                this.disconnect(io.papermc.paper.adventure.PaperAdventure.asVanilla(asyncEvent.kickMessage()), skipLog); // Paper - Adventure // Cheetah - Allow skipping log messages for canceled login events
             }
         }
         return gameprofile; // Paper - Add more fields to AsyncPlayerPreLoginEvent
