From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jendrik Eggers <jendrikeggerskapp@web.de>
Date: Tue, 18 Oct 2022 22:53:50 +0200
Subject: [PATCH] added ProjectileAllowCollideWithPlayerEvent


diff --git a/src/main/java/net/minecraft/world/entity/projectile/Projectile.java b/src/main/java/net/minecraft/world/entity/projectile/Projectile.java
index 6c2d4d6f3a36ab452dfd3c33f66e54f152906639..6daf09689b6b880f5e341e1c6382183058569b06 100644
--- a/src/main/java/net/minecraft/world/entity/projectile/Projectile.java
+++ b/src/main/java/net/minecraft/world/entity/projectile/Projectile.java
@@ -435,7 +435,10 @@ public abstract class Projectile extends Entity implements TraceableEntity {
             if (entity1 instanceof net.minecraft.server.level.ServerPlayer && entity instanceof net.minecraft.server.level.ServerPlayer) {
                 org.bukkit.entity.Player collided = (org.bukkit.entity.Player) entity.getBukkitEntity();
                 org.bukkit.entity.Player shooter = (org.bukkit.entity.Player) entity1.getBukkitEntity();
-                if (!shooter.canSee(collided)) {
+                // Cheetah start - Added ProjectileAllowCollideWithPlayerEvent
+                net.gommehd.cheetah.event.entity.ProjectileAllowCollideWithPlayerEvent allowCollideEvent = org.bukkit.craftbukkit.event.CraftEventFactory.callProjectileAllowCollideWithPlayerEvent(this, entity);
+                if (!allowCollideEvent.isAllowCollide() && !shooter.canSee(collided)) {
+                    // Cheetah end
                     return false;
                 }
             }
diff --git a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
index deba03eb37012c638e08e20cd1c98e9db190c790..d6348437f84e1802124a08463df6aa7b843b3263 100644
--- a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+++ b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
@@ -1473,7 +1473,10 @@ public class CraftEventFactory {
         com.destroystokyo.paper.event.entity.ProjectileCollideEvent event = new com.destroystokyo.paper.event.entity.ProjectileCollideEvent(projectile, collided);
 
         if (projectile.getShooter() instanceof Player && collided instanceof Player) {
-            if (!((Player) projectile.getShooter()).canSee((Player) collided)) {
+            // Cheetah start - Added ProjectileAllowCollideWithPlayerEvent
+            net.gommehd.cheetah.event.entity.ProjectileAllowCollideWithPlayerEvent allowCollideEvent = callProjectileAllowCollideWithPlayerEvent(entity, position.getEntity());
+            if (!allowCollideEvent.isAllowCollide() && !((Player) projectile.getShooter()).canSee((Player) collided)) {
+                // Cheetah end
                 event.setCancelled(true);
                 return event;
             }
@@ -1484,6 +1487,15 @@ public class CraftEventFactory {
     }
     // Paper end
 
+    // Cheetah start - Added ProjectileAllowCollideWithPlayerEvent
+    public static net.gommehd.cheetah.event.entity.ProjectileAllowCollideWithPlayerEvent callProjectileAllowCollideWithPlayerEvent(Entity entity, Entity player) {
+        Projectile projectile = (Projectile) entity.getBukkitEntity();
+        net.gommehd.cheetah.event.entity.ProjectileAllowCollideWithPlayerEvent event = new net.gommehd.cheetah.event.entity.ProjectileAllowCollideWithPlayerEvent(projectile, (Player) player.getBukkitEntity());
+        event.callEvent();
+        return event;
+    }
+    // Cheetah end
+
     public static ProjectileLaunchEvent callProjectileLaunchEvent(Entity entity) {
         Projectile bukkitEntity = (Projectile) entity.getBukkitEntity();
         ProjectileLaunchEvent event = new ProjectileLaunchEvent(bukkitEntity);
