From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: lukas81298 <lukas81298@gommehd.net>
Date: Wed, 17 Aug 2022 22:17:27 +0200
Subject: [PATCH] Updated to latest mojangapi


diff --git a/build.gradle.kts b/build.gradle.kts
index ccc7224a02fd8e9a88beb45275d345660b7a7ecd..1f3158a6aa78d1a0bcb4b0b73ac1c4e7e88886aa 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -9,7 +9,7 @@ plugins {
 dependencies {
     // Cheetah start
     implementation(project(":cheetah-api"))
-    implementation("io.papermc.paper:paper-mojangapi:1.18-R0.1-SNAPSHOT") {
+    implementation("io.papermc.paper:paper-mojangapi:1.19.2-R0.1-SNAPSHOT") {
         exclude("io.papermc.paper", "paper-api")
     }
     // Cheetah end
