From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jendrik Eggers <jendrikeggerskapp@web.de>
Date: Tue, 18 Oct 2022 22:53:50 +0200
Subject: [PATCH] added ProjectileAllowCollideWithPlayerEvent


diff --git a/src/main/java/net/minecraft/world/entity/projectile/Projectile.java b/src/main/java/net/minecraft/world/entity/projectile/Projectile.java
index 1b7cf6d06bdf36f146656727511a461f2520762e..7a96ffc03dd64c17f6d22717b35769b255c99e17 100644
--- a/src/main/java/net/minecraft/world/entity/projectile/Projectile.java
+++ b/src/main/java/net/minecraft/world/entity/projectile/Projectile.java
@@ -4,6 +4,8 @@ import com.google.common.base.MoreObjects;
 import java.util.Iterator;
 import java.util.UUID;
 import javax.annotation.Nullable;
+
+import net.gommehd.cheetah.event.entity.ProjectileAllowCollideWithPlayerEvent;
 import net.minecraft.core.BlockPos;
 import net.minecraft.nbt.CompoundTag;
 import net.minecraft.network.protocol.Packet;
@@ -24,6 +26,7 @@ import net.minecraft.world.phys.EntityHitResult;
 import net.minecraft.world.phys.HitResult;
 import net.minecraft.world.phys.Vec3;
 // CraftBukkit start
+import org.bukkit.craftbukkit.event.CraftEventFactory;
 import org.bukkit.projectiles.ProjectileSource;
 // CraftBukkit end
 
@@ -244,7 +247,8 @@ public abstract class Projectile extends Entity implements TraceableEntity {
             if (entity1 instanceof net.minecraft.server.level.ServerPlayer && entity instanceof net.minecraft.server.level.ServerPlayer) {
                 org.bukkit.entity.Player collided = (org.bukkit.entity.Player) entity.getBukkitEntity();
                 org.bukkit.entity.Player shooter = (org.bukkit.entity.Player) entity1.getBukkitEntity();
-                if (!shooter.canSee(collided)) {
+                ProjectileAllowCollideWithPlayerEvent allowCollideEvent = CraftEventFactory.callProjectileAllowCollideWithPlayerEvent(this, entity); // Cheetah - ProjectileAllowCollideWithPlayerEvent
+                if (!allowCollideEvent.isAllowCollide() && !shooter.canSee(collided)) {
                     return false;
                 }
             }
diff --git a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
index a6edbb5cde568d8138af568122d5d11243f3fb07..adaca2b6695675b9e7fbb97fef2c9b6fdceed1ce 100644
--- a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+++ b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
@@ -14,6 +14,8 @@ import java.util.Map;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 import javax.annotation.Nullable;
+
+import net.gommehd.cheetah.event.entity.ProjectileAllowCollideWithPlayerEvent;
 import net.minecraft.core.BlockPos;
 import net.minecraft.core.Direction;
 import net.minecraft.network.protocol.game.ServerboundContainerClosePacket;
@@ -1428,7 +1430,8 @@ public class CraftEventFactory {
         com.destroystokyo.paper.event.entity.ProjectileCollideEvent event = new com.destroystokyo.paper.event.entity.ProjectileCollideEvent(projectile, collided);
 
         if (projectile.getShooter() instanceof Player && collided instanceof Player) {
-            if (!((Player) projectile.getShooter()).canSee((Player) collided)) {
+            ProjectileAllowCollideWithPlayerEvent allowCollideEvent = callProjectileAllowCollideWithPlayerEvent(entity, position.getEntity()); // Cheetah - ProjectileAllowCollideWithPlayerEvent
+            if (!allowCollideEvent.isAllowCollide() && !((Player) projectile.getShooter()).canSee((Player) collided)) {
                 event.setCancelled(true);
                 return event;
             }
@@ -1439,6 +1442,15 @@ public class CraftEventFactory {
     }
     // Paper end
 
+    // Cheetah start - own ProjectileAllowCollideWithPlayerEvent
+    public static ProjectileAllowCollideWithPlayerEvent callProjectileAllowCollideWithPlayerEvent(Entity entity, Entity player) {
+        Projectile projectile = (Projectile) entity.getBukkitEntity();
+        ProjectileAllowCollideWithPlayerEvent event = new ProjectileAllowCollideWithPlayerEvent(projectile, (Player) player.getBukkitEntity());
+        Bukkit.getPluginManager().callEvent(event);
+        return event;
+    }
+    // Cheetah end
+
     public static ProjectileLaunchEvent callProjectileLaunchEvent(Entity entity) {
         Projectile bukkitEntity = (Projectile) entity.getBukkitEntity();
         ProjectileLaunchEvent event = new ProjectileLaunchEvent(bukkitEntity);
