From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: lukas81298 <lukas81298@gommehd.net>
Date: Thu, 20 Oct 2022 22:50:39 +0200
Subject: [PATCH] support case insensitive commands


diff --git a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index 63c9040ed2349eec500ba6e9090440347c514a3b..aa55ab7804052a72f68dab8bffaf32d6369e06e1 100644
--- a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -2329,10 +2329,21 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Tic
         return map;
     }
 
+    // Cheetah start
+    private static String commandToLower(String command) {
+        if(command.isEmpty()) {
+            return command;
+        }
+        String[] split = command.split(" ");
+        split[0] = split[0].toLowerCase();
+        return String.join(" ", split);
+    }
+    // Cheetah end
+
     private ParseResults<CommandSourceStack> parseCommand(String command) {
         com.mojang.brigadier.CommandDispatcher<CommandSourceStack> com_mojang_brigadier_commanddispatcher = this.server.getCommands().getDispatcher();
 
-        return com_mojang_brigadier_commanddispatcher.parse(command, this.player.createCommandSourceStack());
+        return com_mojang_brigadier_commanddispatcher.parse(commandToLower(command), this.player.createCommandSourceStack()); // Cheetah - added commandToLower()
     }
 
     private boolean tryHandleChat(String message, Instant timestamp, LastSeenMessages.Update acknowledgment) {
diff --git a/src/main/java/org/bukkit/craftbukkit/command/VanillaCommandWrapper.java b/src/main/java/org/bukkit/craftbukkit/command/VanillaCommandWrapper.java
index 6035af2cf08353b3d3801220d8116d8611a0cd37..c7ecfd648142284c2f9884c0dfb5f35af2879c05 100644
--- a/src/main/java/org/bukkit/craftbukkit/command/VanillaCommandWrapper.java
+++ b/src/main/java/org/bukkit/craftbukkit/command/VanillaCommandWrapper.java
@@ -53,7 +53,7 @@ public final class VanillaCommandWrapper extends BukkitCommand {
         Validate.notNull(alias, "Alias cannot be null");
 
         CommandSourceStack icommandlistener = VanillaCommandWrapper.getListener(sender);
-        ParseResults<CommandSourceStack> parsed = this.dispatcher.getDispatcher().parse(this.toDispatcher(args, getName()), icommandlistener);
+        ParseResults<CommandSourceStack> parsed = this.dispatcher.getDispatcher().parse(this.toDispatcher(args, getName().toLowerCase()), icommandlistener); // Cheetah - added to lower case
 
         List<String> results = new ArrayList<>();
         this.dispatcher.getDispatcher().getCompletionSuggestions(parsed).thenAccept((suggestions) -> {
