From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: lukas81298 <lukas81298@gommehd.net>
Date: Thu, 15 Dec 2022 02:20:21 +0100
Subject: [PATCH] do not suppress protocol errors


diff --git a/src/main/java/net/minecraft/network/Connection.java b/src/main/java/net/minecraft/network/Connection.java
index 02833deaa2bb7e5abc655bc1bdbe15c4b3ac7119..a45f3a192ba7505d313c30514fbf48623f6f7f4b 100644
--- a/src/main/java/net/minecraft/network/Connection.java
+++ b/src/main/java/net/minecraft/network/Connection.java
@@ -213,7 +213,7 @@ public class Connection extends SimpleChannelInboundHandler<Packet<?>> {
 
                     if (player != null) player.quitReason = org.bukkit.event.player.PlayerQuitEvent.QuitReason.ERRONEOUS_STATE; // Paper - Add API for quit reason
                     if (flag) {
-                        Connection.LOGGER.debug("Failed to sent packet", throwable);
+                        Connection.LOGGER.warn("Failed to send to " + player.getName() + " packet", throwable); // Cheetah - change log level to warn
                         if (this.getSending() == PacketFlow.CLIENTBOUND) {
                             Packet<?> packet = this.sendLoginDisconnect ? new ClientboundLoginDisconnectPacket(ichatmutablecomponent) : new ClientboundDisconnectPacket(ichatmutablecomponent);
 
