From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: lukas81298 <lukas81298@gommehd.net>
Date: Wed, 5 Apr 2023 22:12:16 +0200
Subject: [PATCH] allow skipping log messages for canceled login events


diff --git a/src/main/java/net/minecraft/server/network/ServerLoginPacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerLoginPacketListenerImpl.java
index 40638b439966619e9c70a18a32abd95b2178fe9f..46d613b6a679222267b16ca6341066df300831ee 100644
--- a/src/main/java/net/minecraft/server/network/ServerLoginPacketListenerImpl.java
+++ b/src/main/java/net/minecraft/server/network/ServerLoginPacketListenerImpl.java
@@ -92,6 +92,7 @@ public class ServerLoginPacketListenerImpl implements ServerLoginPacketListener,
     private ServerPlayer player; // CraftBukkit
     public boolean iKnowThisMayNotBeTheBestIdeaButPleaseDisableUsernameValidation = false; // Paper - username validation overriding
     private int velocityLoginMessageId = -1; // Paper - Add Velocity IP Forwarding Support
+    private boolean disconnectLogged = false; // Cheetah - Allow skipping log messages for canceled login events
 
     public ServerLoginPacketListenerImpl(MinecraftServer server, Connection connection, boolean transferred) {
         this.state = ServerLoginPacketListenerImpl.State.HELLO;
@@ -145,8 +146,15 @@ public class ServerLoginPacketListenerImpl implements ServerLoginPacketListener,
     }
 
     public void disconnect(Component reason) {
+        // Cheetah start - Allow skipping log messages for canceled login events
+        disconnect(reason, false);
+    }
+
+    public void disconnect(Component reason, boolean skipLog) {
         try {
-            ServerLoginPacketListenerImpl.LOGGER.info("Disconnecting {}: {}", this.getUserName(), reason.getString());
+            this.disconnectLogged = true;
+            if(!skipLog) ServerLoginPacketListenerImpl.LOGGER.info("Disconnecting {}: {}", this.getUserName(), reason.getString());
+            // Cheetah end
             this.connection.send(new ClientboundLoginDisconnectPacket(reason));
             this.connection.disconnect(reason);
         } catch (Exception exception) {
@@ -161,7 +169,7 @@ public class ServerLoginPacketListenerImpl implements ServerLoginPacketListener,
 
     @Override
     public void onDisconnect(DisconnectionDetails info) {
-        ServerLoginPacketListenerImpl.LOGGER.info("{} lost connection: {}", this.getUserName(), info.reason().getString());
+        if (!this.disconnectLogged) ServerLoginPacketListenerImpl.LOGGER.info("{} lost connection: {}", this.getUserName(), info.reason().getString()); // Cheetah - Allow skipping log messages for canceled login events
     }
 
     public String getUserName() {
@@ -376,12 +384,14 @@ public class ServerLoginPacketListenerImpl implements ServerLoginPacketListener,
         playerName = gameprofile.getName();
         uniqueId = gameprofile.getId();
         // Paper end - Add more fields to AsyncPlayerPreLoginEvent
+        boolean skipLog = asyncEvent.isSkipLog(); // Cheetah - Allow skipping log messages for canceled login events
 
         if (PlayerPreLoginEvent.getHandlerList().getRegisteredListeners().length != 0) {
             final PlayerPreLoginEvent event = new PlayerPreLoginEvent(playerName, address, uniqueId);
             if (asyncEvent.getResult() != PlayerPreLoginEvent.Result.ALLOWED) {
                 event.disallow(asyncEvent.getResult(), asyncEvent.kickMessage()); // Paper - Adventure
             }
+            event.setSkipLog(skipLog); // Cheetah - Allow skipping log messages for canceled login events
             Waitable<PlayerPreLoginEvent.Result> waitable = new Waitable<PlayerPreLoginEvent.Result>() {
                 @Override
                 protected PlayerPreLoginEvent.Result evaluate() {
@@ -392,11 +402,14 @@ public class ServerLoginPacketListenerImpl implements ServerLoginPacketListener,
 
             ServerLoginPacketListenerImpl.this.server.processQueue.add(waitable);
             if (waitable.get() != PlayerPreLoginEvent.Result.ALLOWED) {
-                this.disconnect(io.papermc.paper.adventure.PaperAdventure.asVanilla(event.kickMessage())); // Paper - Adventure
+                // Cheetah start - Allow skipping log messages for canceled login events
+                skipLog = event.isSkipLog();
+                this.disconnect(io.papermc.paper.adventure.PaperAdventure.asVanilla(event.kickMessage()), skipLog); // Paper - Adventure
+                // Cheetah end
             }
         } else {
             if (asyncEvent.getLoginResult() != AsyncPlayerPreLoginEvent.Result.ALLOWED) {
-                this.disconnect(io.papermc.paper.adventure.PaperAdventure.asVanilla(asyncEvent.kickMessage())); // Paper - Adventure
+                this.disconnect(io.papermc.paper.adventure.PaperAdventure.asVanilla(asyncEvent.kickMessage()), skipLog); // Cheetah - Allow skipping log messages for canceled login events // Paper - Adventure
             }
         }
         return gameprofile; // Paper - Add more fields to AsyncPlayerPreLoginEvent
diff --git a/src/main/java/net/minecraft/server/players/PlayerList.java b/src/main/java/net/minecraft/server/players/PlayerList.java
index d595113472426382f197efff3f9e18a2bbbc3f91..39c998d3ed782a9b65b4ea6f9ccbe6a46e78997a 100644
--- a/src/main/java/net/minecraft/server/players/PlayerList.java
+++ b/src/main/java/net/minecraft/server/players/PlayerList.java
@@ -753,7 +753,7 @@ public abstract class PlayerList {
 
         this.cserver.getPluginManager().callEvent(event);
         if (event.getResult() != PlayerLoginEvent.Result.ALLOWED) {
-            loginlistener.disconnect(io.papermc.paper.adventure.PaperAdventure.asVanilla(event.kickMessage())); // Paper - Adventure
+            loginlistener.disconnect(io.papermc.paper.adventure.PaperAdventure.asVanilla(event.kickMessage()), event.isSkipLog()); // Paper - Adventure
             return null;
         }
         return entity;
