From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jendrik Eggers <jendrikeggerskapp@web.de>
Date: Thu, 13 Apr 2023 17:21:06 +0200
Subject: [PATCH] return false when the allied team is the collide rule team


diff --git a/src/main/java/net/minecraft/world/entity/Entity.java b/src/main/java/net/minecraft/world/entity/Entity.java
index 437956b18d5bffd93c46e0be34cf413e9c704e5a..f52d9bbef20565dff3855d07f71dba24b77c9762 100644
--- a/src/main/java/net/minecraft/world/entity/Entity.java
+++ b/src/main/java/net/minecraft/world/entity/Entity.java
@@ -3270,7 +3270,17 @@ public abstract class Entity implements SyncedDataHolder, Nameable, EntityAccess
     }
 
     public boolean isAlliedTo(Team team) {
-        return this.getTeam() != null ? this.getTeam().isAlliedTo(team) : false;
+        // Cheetah start - Return false when the allied team is the collide rule team
+        final Team ownTeam = this.getTeam();
+        if (ownTeam != null) {
+            if (!io.papermc.paper.configuration.GlobalConfiguration.get().collisions.enablePlayerCollisions &&
+                ownTeam.getName().equals(MinecraftServer.getServer().getPlayerList().collideRuleTeamName)) {
+                return false;
+            }
+            return ownTeam.isAlliedTo(team);
+        }
+        return false;
+        // Cheetah end
     }
 
     // CraftBukkit - start
