From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jendrik Eggers <jendrikeggerskapp@web.de>
Date: Thu, 13 Apr 2023 17:21:06 +0200
Subject: [PATCH] return false when the allied team is the collide rule team


diff --git a/src/main/java/net/minecraft/world/entity/Entity.java b/src/main/java/net/minecraft/world/entity/Entity.java
index 3ffb92a82e17af467afb18c81accc482a93d6a6f..bb6be6821c62ac5136e7f6a3cca41324227a14b6 100644
--- a/src/main/java/net/minecraft/world/entity/Entity.java
+++ b/src/main/java/net/minecraft/world/entity/Entity.java
@@ -3221,7 +3221,17 @@ public abstract class Entity implements Nameable, EntityAccess, CommandSource {
     }
 
     public boolean isAlliedTo(Team team) {
-        return this.getTeam() != null ? this.getTeam().isAlliedTo(team) : false;
+        // Cheetah start
+        final Team ownTeam = this.getTeam();
+        if (ownTeam != null) {
+            if (!io.papermc.paper.configuration.GlobalConfiguration.get().collisions.enablePlayerCollisions &&
+                ownTeam.getName().equals(MinecraftServer.getServer().getPlayerList().collideRuleTeamName)) {
+                return false;
+            }
+            return ownTeam.isAlliedTo(team);
+        }
+        return false;
+        // Cheetah end
     }
 
     // CraftBukkit - start
