From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: lukas81298 <lukas81298@gommehd.net>
Date: Thu, 27 Apr 2023 17:06:53 +0200
Subject: [PATCH] do not break message chain if proxy cancels a chat message


diff --git a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index 1e62ae228e77e6c1f7f1fa90a826fc6dd23c0a35..4105bd3926c06b2d0047db38f8e03c2dab38da3f 100644
--- a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -2172,12 +2172,23 @@ public class ServerGamePacketListenerImpl extends ServerCommonPacketListenerImpl
     }
 
     @Override
-    public void handleChat(ServerboundChatPacket packet) {
+    public void handleChat(ServerboundChatPacket originalPacket) {
         // CraftBukkit start - async chat
         // SPIGOT-3638
         if (this.server.isStopped()) {
             return;
         }
+        // Cheetah start - Do not break message chain if proxy cancels a chat message
+        boolean cancelled = originalPacket.lastSeenMessages().offset() < 0; // Canceling the signed chat packet on bungeecord side will break the chat chain on spigot side. Thus, the server will no longer be able to validate subsequent chat messages.
+                                                                    // We implement a little hack here: If the offset in the lastSeenMessages field is negative, the message will be correctly validated and the chat chain will be updated
+                                                                    // but the server will ignore the chat message, i.e. no events are called, it is not logged, and no player will see the message
+        final ServerboundChatPacket packet;
+        if (cancelled)  {
+            packet = new ServerboundChatPacket(originalPacket.message(), originalPacket.timeStamp(), originalPacket.salt(), originalPacket.signature(), new LastSeenMessages.Update(originalPacket.lastSeenMessages().offset() == Integer.MIN_VALUE ? 0 : -originalPacket.lastSeenMessages().offset(), originalPacket.lastSeenMessages().acknowledged())); // Restore the correct offset (copy packet here, packet itself is immutable record)
+        } else {
+            packet = originalPacket;
+        }
+        // Cheetah end
         // CraftBukkit end
         Optional<LastSeenMessages> optional = this.unpackAndApplyLastSeen(packet.lastSeenMessages());
 
@@ -2192,6 +2203,13 @@ public class ServerGamePacketListenerImpl extends ServerCommonPacketListenerImpl
                     return;
                 }
 
+                // Cheetah start - Do not break message chain if proxy cancels a chat message
+                if (cancelled) {
+                    LOGGER.info("Skipping message of " + player.getScoreboardName() + ": '" + playerchatmessage.signedContent() + " because it was cancelled by BungeeCord.");
+                    return;
+                }
+                // Cheetah end
+
                 CompletableFuture<FilteredText> completablefuture = this.filterTextPacket(playerchatmessage.signedContent()).thenApplyAsync(Function.identity(), this.server.chatExecutor); // CraftBukkit - async chat
                 CompletableFuture<Component> componentFuture = this.server.getChatDecorator().decorate(this.player, null, playerchatmessage.decoratedContent()); // Paper - Adventure
 
@@ -2244,9 +2262,24 @@ public class ServerGamePacketListenerImpl extends ServerCommonPacketListenerImpl
 
     @Override
     public void handleSignedChatCommand(ServerboundChatCommandSignedPacket packet) {
+        // Cheetah start - Do not break message chain if proxy cancels a chat message
+        boolean cancelled = packet.lastSeenMessages().offset() < 0; // Same behavior as for ServerboundChatPacket
+        ServerboundChatCommandSignedPacket packet1;
+        if (cancelled) {
+            packet1 = new ServerboundChatCommandSignedPacket(packet.command(), packet.timeStamp(), packet.salt(), packet.argumentSignatures(), new LastSeenMessages.Update(packet.lastSeenMessages().offset() == Integer.MIN_VALUE ? 0 : -packet.lastSeenMessages().offset(), packet.lastSeenMessages().acknowledged())); // Restore the correct offset (copy packet here, packet itself is immutable record)
+        } else {
+            packet1 = packet;
+        }
+        // Cheetah end
         Optional<LastSeenMessages> optional = this.unpackAndApplyLastSeen(packet.lastSeenMessages());
 
         if (!optional.isEmpty()) {
+            // Cheetah start - Do not break message chain if proxy cancels a chat message
+            if (cancelled) {
+                LOGGER.info("Skipping command of " + player.getScoreboardName() + ": '" + packet1.command() + " because it was cancelled by BungeeCord.");
+                return;
+            }
+            // Cheetah end
             this.tryHandleChat(packet.command(), () -> {
                 // CraftBukkit start - SPIGOT-7346: Prevent disconnected players from executing commands
                 if (this.player.hasDisconnected()) {
