From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: lukas81298 <lukas81298@gommehd.net>
Date: Tue, 31 Jan 2023 23:39:58 +0100
Subject: [PATCH] respect treasure map option for dolphins


diff --git a/src/main/java/net/minecraft/world/entity/animal/Dolphin.java b/src/main/java/net/minecraft/world/entity/animal/Dolphin.java
index 1b1cb0e4d54e52ebe794199e386c54c5d84b3719..d611b2fd8450890c5ab16d3011221c4f710b4e1d 100644
--- a/src/main/java/net/minecraft/world/entity/animal/Dolphin.java
+++ b/src/main/java/net/minecraft/world/entity/animal/Dolphin.java
@@ -158,7 +158,11 @@ public class Dolphin extends WaterAnimal {
     protected void registerGoals() {
         this.goalSelector.addGoal(0, new BreathAirGoal(this));
         this.goalSelector.addGoal(0, new TryFindWaterGoal(this));
-        this.goalSelector.addGoal(1, new Dolphin.DolphinSwimToTreasureGoal(this));
+        // Cheetah start - Respect treasure map option for dolphins
+        if (level().paperConfig().environment.treasureMaps.enabled) {
+            this.goalSelector.addGoal(1, new Dolphin.DolphinSwimToTreasureGoal(this));
+        }
+        // Cheetah end
         this.goalSelector.addGoal(2, new Dolphin.DolphinSwimWithPlayerGoal(this, 4.0D));
         this.goalSelector.addGoal(4, new RandomSwimmingGoal(this, 1.0D, 10));
         this.goalSelector.addGoal(4, new RandomLookAroundGoal(this));
