From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: lukas81298 <lukas81298@gommehd.net>
Date: Tue, 31 Jan 2023 23:39:58 +0100
Subject: [PATCH] respect treasure map option for dolphins


diff --git a/src/main/java/net/minecraft/world/entity/animal/Dolphin.java b/src/main/java/net/minecraft/world/entity/animal/Dolphin.java
index 5af4d590a9b0f17ba53c6959d9c18bd1269878a4..6793c9ea69df5a69c354ec7b096739bfeefcb5b6 100644
--- a/src/main/java/net/minecraft/world/entity/animal/Dolphin.java
+++ b/src/main/java/net/minecraft/world/entity/animal/Dolphin.java
@@ -177,7 +177,11 @@ public class Dolphin extends AgeableWaterCreature {
     protected void registerGoals() {
         this.goalSelector.addGoal(0, new BreathAirGoal(this));
         this.goalSelector.addGoal(0, new TryFindWaterGoal(this));
-        this.goalSelector.addGoal(1, new Dolphin.DolphinSwimToTreasureGoal(this));
+        // Cheetah start - Respect treasure map option for dolphins
+        if (level().paperConfig().environment.treasureMaps.enabled) {
+            this.goalSelector.addGoal(1, new Dolphin.DolphinSwimToTreasureGoal(this));
+        }
+        // Cheetah end
         this.goalSelector.addGoal(2, new Dolphin.DolphinSwimWithPlayerGoal(this, 4.0D));
         this.goalSelector.addGoal(4, new RandomSwimmingGoal(this, 1.0D, 10));
         this.goalSelector.addGoal(4, new RandomLookAroundGoal(this));
