From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Summerfeeling <summerfeelingmc@gmail.com>
Date: Thu, 4 May 2023 12:08:19 +0200
Subject: [PATCH] Added VillagerGiveGiftToHeroEvent


diff --git a/src/main/java/net/minecraft/world/entity/LivingEntity.java b/src/main/java/net/minecraft/world/entity/LivingEntity.java
index f36a075dbee2b96d01899e02460b1d8443e91749..8c00e16e070d90928282f653d2a88c09584989ba 100644
--- a/src/main/java/net/minecraft/world/entity/LivingEntity.java
+++ b/src/main/java/net/minecraft/world/entity/LivingEntity.java
@@ -2018,6 +2018,20 @@ public abstract class LivingEntity extends Entity implements Attackable {
         }
     }
 
+    // Cheetah start - Added VillagerGiveGiftToHeroEvent
+    public List<ItemStack> getFromGiftLootTable(ServerLevel world, ResourceKey<LootTable> lootTableKey) {
+        return this.getFromLootTable(world, lootTableKey, (lootparams_a) -> {
+            return lootparams_a.withParameter(LootContextParams.ORIGIN, this.position()).withParameter(LootContextParams.THIS_ENTITY, this).create(LootContextParamSets.GIFT);
+        });
+    }
+
+    protected List<ItemStack> getFromLootTable(ServerLevel world, ResourceKey<LootTable> lootTableKey, Function<LootParams.Builder, LootParams> lootContextParametersFactory) {
+        LootTable loottable = world.getServer().reloadableRegistries().getLootTable(lootTableKey);
+        LootParams lootparams = (LootParams) lootContextParametersFactory.apply(new LootParams.Builder(world));
+        return loottable.getRandomItems(lootparams);
+    }
+    // Cheetah end - Added VillagerGiveGiftToHeroEvent
+
     public void knockback(double strength, double x, double z) {
         // CraftBukkit start - EntityKnockbackEvent
         this.knockback(strength, x, z, null, io.papermc.paper.event.entity.EntityKnockbackEvent.Cause.UNKNOWN); // Paper - knockback events
diff --git a/src/main/java/net/minecraft/world/entity/ai/behavior/GiveGiftToHero.java b/src/main/java/net/minecraft/world/entity/ai/behavior/GiveGiftToHero.java
index b7b33f76e04d3edf7f42781a5810c81431261acf..efee07834757a16eefde68bb98ab4abff6c97ef9 100644
--- a/src/main/java/net/minecraft/world/entity/ai/behavior/GiveGiftToHero.java
+++ b/src/main/java/net/minecraft/world/entity/ai/behavior/GiveGiftToHero.java
@@ -106,7 +106,19 @@ public class GiveGiftToHero extends Behavior<Villager> {
     }
 
     private void throwGift(ServerLevel world, Villager villager, LivingEntity recipient) {
-        villager.dropFromGiftLootTable(world, getLootTableToThrow(villager), (worldx, stack) -> BehaviorUtils.throwItem(villager, stack, recipient.position()));
+        // Cheetah start - Added VillagerGiveGiftToHeroEvent
+        java.util.List<net.minecraft.world.item.ItemStack> itemStacks = villager.getFromGiftLootTable(world, getLootTableToThrow(villager));
+        java.util.List<org.bukkit.inventory.ItemStack> convertedItemStacks = itemStacks.stream().map(org.bukkit.craftbukkit.inventory.CraftItemStack::asCraftMirror).collect(java.util.stream.Collectors.toList());
+
+        org.bukkit.event.entity.VillagerGiveGiftToHeroEvent event = new org.bukkit.event.entity.VillagerGiveGiftToHeroEvent(villager.getBukkitEntity(), recipient.getBukkitLivingEntity(), convertedItemStacks);
+        event.callEvent();
+
+        if (!event.isCancelled() && !event.getItemStacks().isEmpty()) {
+            itemStacks = event.getItemStacks().stream().map(org.bukkit.craftbukkit.inventory.CraftItemStack::asNMSCopy).toList();
+
+            itemStacks.forEach(stack -> BehaviorUtils.throwItem(villager, stack, recipient.position()));
+        }
+        // Cheetah end - Added VillagerGiveGiftToHeroEvent
     }
 
     private static ResourceKey<LootTable> getLootTableToThrow(Villager villager) {
