From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Summerfeeling <summerfeelingmc@gmail.com>
Date: Thu, 4 May 2023 12:08:19 +0200
Subject: [PATCH] Added VillagerGiveGiftToHeroEvent


diff --git a/src/main/java/net/minecraft/world/entity/ai/behavior/GiveGiftToHero.java b/src/main/java/net/minecraft/world/entity/ai/behavior/GiveGiftToHero.java
index 132d8563db66aab61843613b172179b0fed4fb97..c78e2a2c0dc627227179747f033d62d303d1b45b 100644
--- a/src/main/java/net/minecraft/world/entity/ai/behavior/GiveGiftToHero.java
+++ b/src/main/java/net/minecraft/world/entity/ai/behavior/GiveGiftToHero.java
@@ -24,6 +24,7 @@ import net.minecraft.world.level.storage.loot.LootParams;
 import net.minecraft.world.level.storage.loot.LootTable;
 import net.minecraft.world.level.storage.loot.parameters.LootContextParamSets;
 import net.minecraft.world.level.storage.loot.parameters.LootContextParams;
+import org.bukkit.event.entity.VillagerGiveGiftToHeroEvent; // Cheetah - Added VillagerGiveGiftToHeroEvent
 
 public class GiveGiftToHero extends Behavior<Villager> {
     private static final int THROW_GIFT_AT_DISTANCE = 5;
@@ -104,10 +105,22 @@ public class GiveGiftToHero extends Behavior<Villager> {
     }
 
     private void throwGift(Villager villager, LivingEntity recipient) {
-        for(ItemStack itemStack : this.getItemToThrow(villager)) {
-            BehaviorUtils.throwItem(villager, itemStack, recipient.position());
-        }
+        // Cheetah start - Added VillagerGiveGiftToHeroEvent
+        List<ItemStack> itemStacks = this.getItemToThrow(villager);
+
+        List<org.bukkit.inventory.ItemStack> convertedItemStacks = itemStacks.stream().map(org.bukkit.craftbukkit.inventory.CraftItemStack::asCraftMirror).collect(java.util.stream.Collectors.toList());
+
+        VillagerGiveGiftToHeroEvent event = new VillagerGiveGiftToHeroEvent(villager.getBukkitEntity(), recipient.getBukkitLivingEntity(), convertedItemStacks);
+        event.callEvent();
 
+        if (!event.isCancelled() && !event.getItemStacks().isEmpty()) {
+            itemStacks = event.getItemStacks().stream().map(org.bukkit.craftbukkit.inventory.CraftItemStack::asNMSCopy).toList();
+
+            for (ItemStack itemStack : itemStacks) {
+                BehaviorUtils.throwItem(villager, itemStack, recipient.position());
+            }
+        }
+        // Cheetah end
     }
 
     private List<ItemStack> getItemToThrow(Villager villager) {
