From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: lukas81298 <lukas81298@gommehd.net>
Date: Fri, 9 Jun 2023 18:23:30 +0200
Subject: [PATCH] added option to disable elytra in end cities


diff --git a/src/main/java/io/papermc/paper/configuration/GlobalConfiguration.java b/src/main/java/io/papermc/paper/configuration/GlobalConfiguration.java
index b53b6cc4463675096b061b3b65f14a4695c742e2..8d32d1d5e085311560da720ab820805ae7c4f1bf 100644
--- a/src/main/java/io/papermc/paper/configuration/GlobalConfiguration.java
+++ b/src/main/java/io/papermc/paper/configuration/GlobalConfiguration.java
@@ -333,6 +333,7 @@ public class GlobalConfiguration extends ConfigurationPart {
         public boolean useDimensionTypeForCustomSpawners = false;
         public boolean strictAdvancementDimensionCheck = false;
         public IntOr.Default compressionLevel = IntOr.Default.USE_DEFAULT;
+        public boolean enableElytraInEndCities = true;
     }
 
     public BlockUpdates blockUpdates;
diff --git a/src/main/java/net/minecraft/world/level/levelgen/structure/structures/EndCityPieces.java b/src/main/java/net/minecraft/world/level/levelgen/structure/structures/EndCityPieces.java
index d859c1ce8d60920669d9d1afb3b71692375800a8..3682283cf32b9e7ec6adb4b9b1c30fed599eb08f 100644
--- a/src/main/java/net/minecraft/world/level/levelgen/structure/structures/EndCityPieces.java
+++ b/src/main/java/net/minecraft/world/level/levelgen/structure/structures/EndCityPieces.java
@@ -299,7 +299,7 @@ public class EndCityPieces {
                         entityshulker.setPos((double) pos.getX() + 0.5D, (double) pos.getY(), (double) pos.getZ() + 0.5D);
                         world.addFreshEntity(entityshulker);
                     }
-                } else if (metadata.startsWith("Elytra")) {
+                } else if (metadata.startsWith("Elytra") && io.papermc.paper.configuration.GlobalConfiguration.get().misc.enableElytraInEndCities) {
                     ItemFrame entityitemframe = new ItemFrame(world.getLevel(), pos, this.placeSettings.getRotation().rotate(Direction.SOUTH));
 
                     entityitemframe.setItem(new ItemStack(Items.ELYTRA), false);
