From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: lukas81298 <lukas81298@gommehd.net>
Date: Fri, 9 Jun 2023 18:23:30 +0200
Subject: [PATCH] added option to disable elytra in end cities


diff --git a/src/main/java/io/papermc/paper/configuration/GlobalConfiguration.java b/src/main/java/io/papermc/paper/configuration/GlobalConfiguration.java
index 690086adcb1f87adc0b3ab664fde735ad3e1093a..8792dd1d6a8d37a535048f184f127941fe16399d 100644
--- a/src/main/java/io/papermc/paper/configuration/GlobalConfiguration.java
+++ b/src/main/java/io/papermc/paper/configuration/GlobalConfiguration.java
@@ -347,6 +347,7 @@ public class GlobalConfiguration extends ConfigurationPart {
         public IntOr.Default compressionLevel = IntOr.Default.USE_DEFAULT;
         @Comment("Defines the leniency distance added on the server to the interaction range of a player when validating interact packets.")
         public DoubleOr.Default clientInteractionLeniencyDistance = DoubleOr.Default.USE_DEFAULT;
+        public boolean enableElytraInEndCities = true; // Cheetah - Added option to disable elytra in end cities
     }
 
     public BlockUpdates blockUpdates;
diff --git a/src/main/java/net/minecraft/world/level/levelgen/structure/structures/EndCityPieces.java b/src/main/java/net/minecraft/world/level/levelgen/structure/structures/EndCityPieces.java
index 4f77ccd9b93576a7ac6dd29b8a9e7b0390e43513..a895ba7d753576b284f4361076cc0336df07c9d1 100644
--- a/src/main/java/net/minecraft/world/level/levelgen/structure/structures/EndCityPieces.java
+++ b/src/main/java/net/minecraft/world/level/levelgen/structure/structures/EndCityPieces.java
@@ -299,7 +299,7 @@ public class EndCityPieces {
                         entityshulker.setPos((double) pos.getX() + 0.5D, (double) pos.getY(), (double) pos.getZ() + 0.5D);
                         world.addFreshEntity(entityshulker);
                     }
-                } else if (metadata.startsWith("Elytra")) {
+                } else if (metadata.startsWith("Elytra") && io.papermc.paper.configuration.GlobalConfiguration.get().misc.enableElytraInEndCities) { // Cheetah - Added option to disable elytra in end cities
                     ItemFrame entityitemframe = new ItemFrame(world.getLevel(), pos, this.placeSettings.getRotation().rotate(Direction.SOUTH));
 
                     entityitemframe.setItem(new ItemStack(Items.ELYTRA), false);
