From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: lukas81298 <lukas81298@gommehd.net>
Date: Fri, 9 Jun 2023 18:23:30 +0200
Subject: [PATCH] added option to disable elytra in end cities


diff --git a/src/main/java/io/papermc/paper/configuration/GlobalConfiguration.java b/src/main/java/io/papermc/paper/configuration/GlobalConfiguration.java
index f2e30d8743a97c0541808dc5c03300c0d881b843..a27d74dddbbaae59a800d167e8cbaad54e84af0d 100644
--- a/src/main/java/io/papermc/paper/configuration/GlobalConfiguration.java
+++ b/src/main/java/io/papermc/paper/configuration/GlobalConfiguration.java
@@ -334,6 +334,7 @@ public class GlobalConfiguration extends ConfigurationPart {
         public boolean useDimensionTypeForCustomSpawners = false;
         public boolean strictAdvancementDimensionCheck = false;
         public IntOr.Default compressionLevel = IntOr.Default.USE_DEFAULT;
+        public boolean enableElytraInEndCities = true; // Cheetah - Added option to disable elytra in end cities
     }
 
     public BlockUpdates blockUpdates;
diff --git a/src/main/java/net/minecraft/world/level/levelgen/structure/structures/EndCityPieces.java b/src/main/java/net/minecraft/world/level/levelgen/structure/structures/EndCityPieces.java
index d859c1ce8d60920669d9d1afb3b71692375800a8..b2b3b4eb6e29a376b14da1a79e0d73cc78f9f526 100644
--- a/src/main/java/net/minecraft/world/level/levelgen/structure/structures/EndCityPieces.java
+++ b/src/main/java/net/minecraft/world/level/levelgen/structure/structures/EndCityPieces.java
@@ -299,7 +299,7 @@ public class EndCityPieces {
                         entityshulker.setPos((double) pos.getX() + 0.5D, (double) pos.getY(), (double) pos.getZ() + 0.5D);
                         world.addFreshEntity(entityshulker);
                     }
-                } else if (metadata.startsWith("Elytra")) {
+                } else if (metadata.startsWith("Elytra") && io.papermc.paper.configuration.GlobalConfiguration.get().misc.enableElytraInEndCities) { // Cheetah - Added option to disable elytra in end cities
                     ItemFrame entityitemframe = new ItemFrame(world.getLevel(), pos, this.placeSettings.getRotation().rotate(Direction.SOUTH));
 
                     entityitemframe.setItem(new ItemStack(Items.ELYTRA), false);
