From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: lukas81298 <lukas81298@gommehd.net>
Date: Fri, 9 Jun 2023 18:24:40 +0200
Subject: [PATCH] Async-able StructuresLocateEvent


diff --git a/src/main/java/net/minecraft/world/level/chunk/ChunkGenerator.java b/src/main/java/net/minecraft/world/level/chunk/ChunkGenerator.java
index 0708aaa7d25c674ab2ce431a262fed2459fd633d..4d918a1406d9a986b665f13b941108fa6a82bea7 100644
--- a/src/main/java/net/minecraft/world/level/chunk/ChunkGenerator.java
+++ b/src/main/java/net/minecraft/world/level/chunk/ChunkGenerator.java
@@ -126,7 +126,7 @@ public abstract class ChunkGenerator {
         final org.bukkit.Location origin = io.papermc.paper.util.MCUtil.toLocation(world, center);
         final List<org.bukkit.generator.structure.Structure> apiStructures = structures.stream().map(Holder::value).map(nms -> org.bukkit.craftbukkit.generator.structure.CraftStructure.minecraftToBukkit(nms)).toList();
         if (!apiStructures.isEmpty()) {
-            final io.papermc.paper.event.world.StructuresLocateEvent event = new io.papermc.paper.event.world.StructuresLocateEvent(bukkitWorld, origin, apiStructures, radius, skipReferencedStructures);
+            final io.papermc.paper.event.world.StructuresLocateEvent event = new io.papermc.paper.event.world.StructuresLocateEvent(bukkitWorld, origin, apiStructures, radius, skipReferencedStructures, !io.papermc.paper.util.MCUtil.isMainThread()); // Cheetah - added async parameter
             if (!event.callEvent()) {
                 return null;
             }
