From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: lukas81298 <lukas81298@gommehd.net>
Date: Thu, 21 Sep 2023 14:46:51 +0200
Subject: [PATCH] removed test server


diff --git a/build.gradle.kts b/build.gradle.kts
index 88f823950c2de3257eb92e8375f766acbb7a2627..74d220acf7491a862265d19f49944a35594d4bd9 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -214,9 +214,3 @@ val runtimeClasspathForRunDev = sourceSets.main.flatMap { src ->
         elements.filterNot { file -> file.asFile.endsWith("minecraft.jar") }
     }
 }
-tasks.registerRunTask("runDev") {
-    description = "Spin up a non-relocated Mojang-mapped test server"
-    classpath(tasks.filterProjectDir.flatMap { it.outputJar })
-    classpath(runtimeClasspathForRunDev)
-    jvmArgs("-DPaper.isRunDev=true")
-}
