From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jendrik Eggers <jendrikeggerskapp@web.de>
Date: Sun, 28 Apr 2024 15:02:31 +0200
Subject: [PATCH] ignore javadoc errors


diff --git a/build.gradle.kts b/build.gradle.kts
index 0f5adb7190734a8c48a50f07714a70b9d422fa29..ebad9821f37000174168918b105c7a04ac36a60a 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -111,6 +111,12 @@ tasks.compileTestJava {
 }
 // Paper end
 
+// Cheetah start - ignore javadoc errors
+tasks.withType<Javadoc> {
+    (options as StandardJavadocDocletOptions).addStringOption("Xdoclint:none", "-quiet") // Cheetah - ignore javadoc errors
+}
+// Cheetah end - ignore javadoc errors
+
 publishing {
     publications.create<MavenPublication>("maven") {
     }
