From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jendrik Eggers <jendrikeggerskapp@web.de>
Date: Sat, 21 Oct 2023 23:56:05 +0200
Subject: [PATCH] implement performance improvements from upstream 1.20.2


diff --git a/src/main/java/net/minecraft/server/level/ChunkMap.java b/src/main/java/net/minecraft/server/level/ChunkMap.java
index fb6cbcc4839aef7dc4bd4a49613f892b07aab353..fbb082682626c442acfcb4470209e7e1590c0f69 100644
--- a/src/main/java/net/minecraft/server/level/ChunkMap.java
+++ b/src/main/java/net/minecraft/server/level/ChunkMap.java
@@ -1568,7 +1568,7 @@ public class ChunkMap extends ChunkStorage implements ChunkHolder.PlayerProvider
                 // Paper end - check Y
 
                 // CraftBukkit start - respect vanish API
-                if (!player.getBukkitEntity().canSee(this.entity.getBukkitEntity())) {
+                if (flag && !player.getBukkitEntity().canSee(this.entity.getBukkitEntity())) { // Cheetah - only consider hits
                     flag = false;
                 }
                 // CraftBukkit end
