From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jendrik Eggers <jendrikeggerskapp@web.de>
Date: Sun, 28 Apr 2024 15:02:31 +0200
Subject: [PATCH] ignore javadoc errors


diff --git a/build.gradle.kts b/build.gradle.kts
index edbe0313184cd099c8b47d0d34af4dcaf6b3d516..d31d2a0a41f71004c8ddd71568ca4bda6de444db 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -126,6 +126,12 @@ tasks.compileTestJava {
 }
 // Paper end
 
+// Cheetah start - ignore javadoc errors
+tasks.withType<Javadoc> {
+    (options as StandardJavadocDocletOptions).addStringOption("Xdoclint:none", "-quiet") // Cheetah - ignore javadoc errors
+}
+// Cheetah end - ignore javadoc errors
+
 publishing {
     publications.create<MavenPublication>("maven") {
     }
