From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Dominik <yt.dominik48n@gmail.com>
Date: Sun, 11 Aug 2024 09:37:49 +0200
Subject: [PATCH] Fix GLOBAL-1080 (Paper #9350)


diff --git a/src/main/java/net/minecraft/server/level/ChunkMap.java b/src/main/java/net/minecraft/server/level/ChunkMap.java
index cfeeddf2cb4ff50dbc29c6913e78ca1dee076790..960b28620ae4fc75e61be73eddcd4f1bc7d4e28d 100644
--- a/src/main/java/net/minecraft/server/level/ChunkMap.java
+++ b/src/main/java/net/minecraft/server/level/ChunkMap.java
@@ -1288,6 +1288,8 @@ public class ChunkMap extends ChunkStorage implements ChunkHolder.PlayerProvider
                 flag = flag && this.entity.broadcastToPlayer(player) && ChunkMap.this.isChunkTracked(player, this.entity.chunkPosition().x, this.entity.chunkPosition().z);
                 // Paper end - Configurable entity tracking range by Y
 
+                flag |= player.getCamera() == this.entity; // Cheetah - Fix GLOBAL-1080
+
                 // CraftBukkit start - respect vanish API
                 if (flag && !player.getBukkitEntity().canSee(this.entity.getBukkitEntity())) { // Paper - only consider hits
                     flag = false;
diff --git a/src/main/java/net/minecraft/server/level/ServerPlayer.java b/src/main/java/net/minecraft/server/level/ServerPlayer.java
index fc7f7a34babd095a51b5321f600aef65a2a9d123..59bf64335563e0c995785df12196572ff075177e 100644
--- a/src/main/java/net/minecraft/server/level/ServerPlayer.java
+++ b/src/main/java/net/minecraft/server/level/ServerPlayer.java
@@ -1010,8 +1010,23 @@ public class ServerPlayer extends net.minecraft.world.entity.player.Player imple
 
         if (entity != this) {
             if (entity.isAlive()) {
-                this.absMoveTo(entity.getX(), entity.getY(), entity.getZ(), entity.getYRot(), entity.getXRot());
-                this.serverLevel().getChunkSource().move(this);
+                // Cheetah start - Fix GLOBAL-1080
+                if (entity.level() != this.level()) {
+                    this.getBukkitEntity().teleport(entity.getBukkitEntity().getLocation(), TeleportCause.SPECTATE);
+
+                    ServerLevel serverLevel = (ServerLevel) entity.level();
+                    Optional.ofNullable(serverLevel.getChunkSource().chunkMap.entityMap.get(entity.getId())).ifPresent(trackedEntity -> trackedEntity.updatePlayer(this));
+
+                    this.connection.send(new ClientboundSetCameraPacket(entity));
+                } else {
+                    if (!entity.chunkPosition().equals(this.chunkPosition())) {
+                        this.connection.internalTeleport(PositionMoveRotation.of(entity), java.util.Collections.emptySet());
+                    }
+
+                    this.absMoveTo(entity.getX(), entity.getY(), entity.getZ(), entity.getYRot(), entity.getXRot());
+                    this.serverLevel().getChunkSource().move(this);
+                }
+                // Cheetah end
                 if (this.wantsToStopRiding()) {
                     this.setCamera(this);
                 }
