From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Dominik <yt.dominik48n@gmail.com>
Date: Sun, 11 Aug 2024 09:37:49 +0200
Subject: [PATCH] Fix GLOBAL-1080 (Paper #9350)


diff --git a/src/main/java/net/minecraft/server/level/ChunkMap.java b/src/main/java/net/minecraft/server/level/ChunkMap.java
index 5b3a886c624b36557cbfaccdc3fb05a46a4ba36a..6fca35d9d39194f77df2c9c09667e11f947a98ee 100644
--- a/src/main/java/net/minecraft/server/level/ChunkMap.java
+++ b/src/main/java/net/minecraft/server/level/ChunkMap.java
@@ -1294,6 +1294,8 @@ public class ChunkMap extends ChunkStorage implements ChunkHolder.PlayerProvider
                 flag = flag && this.entity.broadcastToPlayer(player) && ChunkMap.this.isChunkTracked(player, this.entity.chunkPosition().x, this.entity.chunkPosition().z);
                 // Paper end - Configurable entity tracking range by Y
 
+                flag |= player.getCamera() == this.entity; // Cheetah - Fix GLOBAL-1080
+
                 // CraftBukkit start - respect vanish API
                 if (flag && !player.getBukkitEntity().canSee(this.entity.getBukkitEntity())) { // Paper - only consider hits
                     flag = false;
diff --git a/src/main/java/net/minecraft/server/level/ServerPlayer.java b/src/main/java/net/minecraft/server/level/ServerPlayer.java
index cffbd3300967e5d80b5973b35a76235bb2aa1b73..28b87a47f1b447c3fb34b292516ce68ebea82650 100644
--- a/src/main/java/net/minecraft/server/level/ServerPlayer.java
+++ b/src/main/java/net/minecraft/server/level/ServerPlayer.java
@@ -1013,8 +1013,23 @@ public class ServerPlayer extends net.minecraft.world.entity.player.Player imple
 
         if (entity != this) {
             if (entity.isAlive()) {
-                this.absMoveTo(entity.getX(), entity.getY(), entity.getZ(), entity.getYRot(), entity.getXRot());
-                this.serverLevel().getChunkSource().move(this);
+                // Cheetah start - Fix GLOBAL-1080
+                if (entity.level() != this.level()) {
+                    this.getBukkitEntity().teleport(entity.getBukkitEntity().getLocation(), TeleportCause.SPECTATE);
+
+                    ServerLevel serverLevel = (ServerLevel) entity.level();
+                    Optional.ofNullable(serverLevel.getChunkSource().chunkMap.entityMap.get(entity.getId())).ifPresent(trackedEntity -> trackedEntity.updatePlayer(this));
+
+                    this.connection.send(new ClientboundSetCameraPacket(entity));
+                } else {
+                    if (!entity.chunkPosition().equals(this.chunkPosition())) {
+                        this.connection.internalTeleport(entity.getX(), entity.getY(), entity.getZ(), entity.getYRot(), entity.getXRot(), java.util.Collections.emptySet());
+                    }
+
+                    this.absMoveTo(entity.getX(), entity.getY(), entity.getZ(), entity.getYRot(), entity.getXRot());
+                    this.serverLevel().getChunkSource().move(this);
+                }
+                // Cheetah end
                 if (this.wantsToStopRiding()) {
                     this.setCamera(this);
                 }
