From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jendrik Eggers <jendrikeggerskapp@web.de>
Date: Mon, 28 Oct 2024 17:04:30 +0100
Subject: [PATCH] fix NaN delta movement


diff --git a/src/main/java/net/minecraft/world/entity/projectile/AbstractHurtingProjectile.java b/src/main/java/net/minecraft/world/entity/projectile/AbstractHurtingProjectile.java
index 19ff16e1ba406584f3cdd760d0269a50980b0a26..00f3c8b9db31f85da190ab4554dbfeb6b1e442dd 100644
--- a/src/main/java/net/minecraft/world/entity/projectile/AbstractHurtingProjectile.java
+++ b/src/main/java/net/minecraft/world/entity/projectile/AbstractHurtingProjectile.java
@@ -93,6 +93,13 @@ public abstract class AbstractHurtingProjectile extends Projectile {
                 this.igniteForSeconds(1.0F);
             }
 
+            // Cheetah start - fix NaN delta movement
+            final Vec3 deltaMovement = this.getDeltaMovement();
+            if (Double.isNaN(deltaMovement.x) && Double.isNaN(deltaMovement.y) && Double.isNaN(deltaMovement.z)) {
+                this.setDeltaMovement(Vec3.ZERO);
+            }
+            // Cheetah end - fix NaN delta movement
+
             if (movingobjectposition.getType() != HitResult.Type.MISS && this.isAlive()) {
                 this.preHitTargetOrDeflectSelf(movingobjectposition); // CraftBukkit - projectile hit event
             }
