From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jendrik Eggers <jendrikeggerskapp@web.de>
Date: Mon, 28 Oct 2024 17:04:30 +0100
Subject: [PATCH] fix NaN delta movement


diff --git a/src/main/java/net/minecraft/world/entity/projectile/AbstractHurtingProjectile.java b/src/main/java/net/minecraft/world/entity/projectile/AbstractHurtingProjectile.java
index f79fc0ed58af2cf55e8642b6d7fc9e4f57d0ba20..cbe515fc0b2c092be5071cc0677f3dde7eb1b332 100644
--- a/src/main/java/net/minecraft/world/entity/projectile/AbstractHurtingProjectile.java
+++ b/src/main/java/net/minecraft/world/entity/projectile/AbstractHurtingProjectile.java
@@ -83,6 +83,13 @@ public abstract class AbstractHurtingProjectile extends Projectile {
                 this.igniteForSeconds(1.0F);
             }
 
+            // Cheetah start - fix NaN delta movement
+            final Vec3 deltaMovement = this.getDeltaMovement();
+            if (Double.isNaN(deltaMovement.x) && Double.isNaN(deltaMovement.y) && Double.isNaN(deltaMovement.z)) {
+                this.setDeltaMovement(Vec3.ZERO);
+            }
+            // Cheetah end - fix NaN delta movement
+
             HitResult movingobjectposition = ProjectileUtil.getHitResultOnMoveVector(this, this::canHitEntity, this.getClipType());
 
             if (movingobjectposition.getType() != HitResult.Type.MISS) {
