From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jendrik Eggers <jendrikeggerskapp@web.de>
Date: Mon, 28 Oct 2024 17:04:30 +0100
Subject: [PATCH] fix NaN delta movement


diff --git a/src/main/java/net/minecraft/world/entity/projectile/AbstractHurtingProjectile.java b/src/main/java/net/minecraft/world/entity/projectile/AbstractHurtingProjectile.java
index a84dccc59f9dc5298b18f263dc779c30c9b7ddab..311813a2e78070eed895eeb793836fbdb5cdc76b 100644
--- a/src/main/java/net/minecraft/world/entity/projectile/AbstractHurtingProjectile.java
+++ b/src/main/java/net/minecraft/world/entity/projectile/AbstractHurtingProjectile.java
@@ -97,6 +97,13 @@ public abstract class AbstractHurtingProjectile extends Projectile {
                 this.igniteForSeconds(1.0F);
             }
 
+            // Cheetah start - fix NaN delta movement
+            final Vec3 deltaMovement = this.getDeltaMovement();
+            if (Double.isNaN(deltaMovement.x) && Double.isNaN(deltaMovement.y) && Double.isNaN(deltaMovement.z)) {
+                this.setDeltaMovement(Vec3.ZERO);
+            }
+            // Cheetah end - fix NaN delta movement
+
             if (movingobjectposition.getType() != HitResult.Type.MISS && this.isAlive()) {
                 this.preHitTargetOrDeflectSelf(movingobjectposition); // CraftBukkit - projectile hit event
             }
