From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: lennoxlotl <admin@lennoxlotl.dev>
Date: Mon, 6 Jan 2025 02:26:46 +0100
Subject: [PATCH] fix: shield sounds not playing for other players


diff --git a/src/main/java/net/minecraft/world/entity/LivingEntity.java b/src/main/java/net/minecraft/world/entity/LivingEntity.java
index 2aa6374cd4a96efd85899be8cd3172a8257bfe6b..a868594f371fea25429c0d20d2f788407b8bc218 100644
--- a/src/main/java/net/minecraft/world/entity/LivingEntity.java
+++ b/src/main/java/net/minecraft/world/entity/LivingEntity.java
@@ -1428,6 +1428,14 @@ public abstract class LivingEntity extends Entity implements Attackable {
             float f1 = amount; final float originalAmount = f1; // Paper - revert to vanilla #hurt - OBFHELPER
             boolean flag = amount > 0.0F && this.isDamageSourceBlocked(source); // Copied from below
             float f2 = 0.0F;
+            // Cheetah start - Play shield sounds for all players
+            boolean shieldBroken;
+            if (source.getDirectEntity() instanceof LivingEntity livingEntity) {
+                shieldBroken = livingEntity.canDisableShield();
+            } else {
+                shieldBroken = false;
+            }
+            // Cheetah end
 
             // CraftBukkit - Moved into handleEntityDamage(DamageSource, float) for get f and actuallyHurt(DamageSource, float, EntityDamageEvent) for handle damage
             if (false && amount > 0.0F && this.isDamageSourceBlocked(source)) {
@@ -1538,6 +1546,22 @@ public abstract class LivingEntity extends Entity implements Attackable {
             if (flag1) {
                 if (flag) {
                     this.level().broadcastEntityEvent(this, (byte) 29);
+                    // Cheetah start - Play shield sounds for all players
+                    level().players().stream()
+                        .filter(entity -> entity != this)
+                        .filter(entity -> entity instanceof net.minecraft.server.level.ServerPlayer)
+                        .forEach(entity ->
+                            ((net.minecraft.server.level.ServerPlayer) entity).connection.sendPacket(
+                                new net.minecraft.network.protocol.game.ClientboundSoundPacket(
+                                    Holder.direct(shieldBroken ? SoundEvents.SHIELD_BREAK : SoundEvents.SHIELD_BLOCK),
+                                    getSoundSource(),
+                                    position().x,
+                                    position().y,
+                                    position().z,
+                                    shieldBroken ? 0.8F : 1.0F,
+                                    0.8F + level().random.nextFloat() * 0.4F,
+                                    level().random.nextLong())));
+                    // Cheetah end
                 } else {
                     this.level().broadcastDamageEvent(this, source);
                 }
@@ -1586,7 +1610,7 @@ public abstract class LivingEntity extends Entity implements Attackable {
                     this.die(source);
                     this.silentDeath = false; // Paper - cancellable death event - reset to default
                 }
-            } else if (flag1) {
+            } else if (flag1 && !flag) { // Cheetah - Play shield sounds for all players
                 this.playHurtSound(source);
             }
 
