--- a/net/minecraft/server/players/PlayerList.java
+++ b/net/minecraft/server/players/PlayerList.java
@@ -306,9 +_,8 @@
         // Paper end - Configurable player collision
         // CraftBukkit start - moved down
         LOGGER.info(
-            "{}[{}] logged in with entity id {} at ([{}]{}, {}, {})", // CraftBukkit - add world name
+            "{} logged in with entity id {} at ([{}]{}, {}, {})", // CraftBukkit - add world name // Cheetah - removed ip address
             player.getPlainTextName(),
-            loggableAddress,
             player.getId(),
             serverLevel.serverLevelData.getLevelName(), // CraftBukkit - add world name
             player.getX(),
