--- a/net/minecraft/server/players/PlayerList.java
+++ b/net/minecraft/server/players/PlayerList.java
@@ -314,9 +_,8 @@
         // Paper end - Configurable player collision
         // CraftBukkit start - moved down
         LOGGER.info(
-            "{}[{}] logged in with entity id {} at ([{}]{}, {}, {})", // Paper - add world identifier
+            "{} logged in with entity id {} at ([{}]{}, {}, {})", // Paper - add world identifier // Cheetah - removed ip address
             player.getPlainTextName(),
-            address,
             player.getId(),
             level.dimension().identifier(), // Paper - add world identifier
             player.getX(),
